<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Nhi Hin" />

<meta name="date" content="2020-05-04" />

<title>Import 5XFAD Mouse Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mousevsfish</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Import 5XFAD Mouse Data</h1>
<h4 class="author">Nhi Hin</h4>
<h4 class="date">2020-05-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mousevsfish/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200504code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200504)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200504code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200504)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnhihinmousevsfishtree744a3be8cae81ae7cace398e88788ee38afaac73targetblank744a3bea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/nhihin/mousevsfish/tree/744a3be8cae81ae7cace398e88788ee38afaac73" target="_blank">744a3be</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomnhihinmousevsfishtree744a3be8cae81ae7cace398e88788ee38afaac73targetblank744a3bea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/nhihin/mousevsfish/tree/744a3be8cae81ae7cace398e88788ee38afaac73" target="_blank">744a3be</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/de/
    Ignored:    data/enrichment/

Untracked files:
    Untracked:  analysis/import-data.Rmd
    Untracked:  data/datasets/
    Untracked:  notes.docx
    Untracked:  ~$notes.docx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>0. Introduction</h2>
<ul>
<li>In this file we will import in the existing publically available mouse 5XFAD RNA-seq datasets directly from GEO and have a look at the datasets. If the PCAs look decent then I will re-process the raw fastqs to be consistent with our zebrafish dataset.</li>
</ul>
</div>
<div id="unknown-age-mouse-hippocampus-5xfad-and-wt" class="section level2">
<h2>1. Unknown age, mouse hippocampus, 5XFAD and WT</h2>
<ul>
<li>The publication is not out yet, so there aren’t many details. Age of mice is unknown, so we can’t use this dataset for now. Will email authors to ask them for the age.</li>
</ul>
<pre class="r"><code>geo &lt;- getGEO(&quot;GSE115437&quot;, GSEMatrix = TRUE)
geo$GSE115437_series_matrix.txt.gz@phenoData %&gt;% pData %&gt;% as_tibble</code></pre>
</div>
<div id="and-4-months-mouse-hippocampus-5xfad-only" class="section level2">
<h2>2. <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE97113">1,2 and 4 months, mouse hippocampus, 5XFAD only</a></h2>
<ul>
<li><p>They don’t have WT controls in this dataset, although it’s the only one I could find that includes these earlier age groups which are likely to be more comparable to our zebrafish.</p></li>
<li><p>Will try seeing where they lie on a PCA with other 5XFAD datasets to see if it’s possible to combine them or generally how comparable they are.</p></li>
<li><p><strong>Platform</strong>: Illumina HiSeq 2500</p></li>
</ul>
<pre class="r"><code># Retrieve the sample metadata from GEO
geo &lt;- getGEO(&quot;GSE97113&quot;, GSEMatrix = TRUE)</code></pre>
<pre><code>Found 1 file(s)</code></pre>
<pre><code>GSE97113_series_matrix.txt.gz</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<pre><code>File stored at: </code></pre>
<pre><code>/var/folders/mf/j7h9p8l55k9fb0rgm_f753cc0000gn/T//RtmpzchMFb/GPL17021.soft</code></pre>
<pre class="r"><code>samples &lt;- geo$GSE97113_series_matrix.txt.gz@phenoData %&gt;%
  pData %&gt;% 
  as_tibble %&gt;% 
  dplyr::select(title, description,
                geo_accession,
                starts_with(&quot;characteristics&quot;)) %&gt;%
  dplyr::rename(genotype = characteristics_ch1,
                id = description,
                sex = characteristics_ch1.1,
                age = characteristics_ch1.2,
                tissue = characteristics_ch1.3) %&gt;%
  dplyr::mutate(genotype = &quot;5XFAD&quot; %&gt;% as.factor,
                sex = gsub(x=sex, pattern=&quot;Sex: &quot;, replacement = &quot;&quot;) %&gt;% 
                  as.factor,
                age = gsub(x=age, pattern=&quot;age: | Month&quot;, replacement = &quot;&quot;) %&gt;% 
                  as.factor,
                tissue = &quot;Hippocampus&quot; %&gt;% as.factor,
                title = gsub(x=title,pattern=&quot; &quot;, replacement=&quot;_&quot;))

# Downloaded counts from GEO
counts &lt;- read_csv(here(&quot;data/datasets/mouse/GSE97113_5XFAD_counts.csv&quot;))</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_double(),
  Gene.ID = col_character(),
  Associated.Gene.Name = col_character(),
  Description = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<pre class="r"><code>genes &lt;- counts %&gt;% dplyr::select(Gene.ID, Associated.Gene.Name, Description)
counts &lt;- counts %&gt;% as.data.frame %&gt;% column_to_rownames(&quot;Gene.ID&quot;) %&gt;% 
  dplyr::select(-Associated.Gene.Name, -Description)

colnames(counts)==samples$id # Check ordering of samples is the same</code></pre>
<pre><code> [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[16] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code># Set the sample names in counts matrix to readable name
colnames(counts)=samples$title</code></pre>
<ul>
<li>Create DGEList to store counts, samples, and genes.</li>
</ul>
<pre class="r"><code>dge_GSE97113 &lt;- DGEList(counts=counts, 
                        samples = samples,
                        genes = genes, remove.zeros = TRUE)</code></pre>
<pre><code>Removing 11810 rows with all zero counts</code></pre>
<ul>
<li>Quick PCA to check samples and their similarity to each other.</li>
</ul>
<pre class="r"><code>pca &lt;- prcomp(t(cpm(dge_GSE97113, log = TRUE)))

shape_factor &lt;- paste0(samples$sex, &quot;_&quot;, samples$age, &quot;_mths&quot;) %&gt;% 
  factor(levels = unique(.))

pca_plot_GSE97113 &lt;- pca$x %&gt;% magrittr::extract(, c(&quot;PC1&quot;, &quot;PC2&quot;)) %&gt;%
  set_colnames(c(&quot;PCa&quot;, &quot;PCb&quot;)) %&gt;%
  as.data.frame %&gt;%
  rownames_to_column(&quot;samples&quot;) %&gt;%
  left_join((dge_GSE97113$samples %&gt;% rownames_to_column(&quot;samples&quot;)), by=&quot;samples&quot;) %&gt;%
  ggplot(aes(x=PCa, y = PCb, shape = sex, colour = age)) +
  geom_point(shape = 1,size = 3.3)+
  geom_point(alpha = 1,size=3) + 
  scale_shape_manual(values = c(1,16)) + 
  #scale_shape_manual(values = c(0,15,1,16,2,17),
  #                   labels = levels(shape_factor)) +
  labs(x = &quot;Principal Component 1 (11.5%)&quot;, 
       y = &quot;Principal Component 2 (9.4%)&quot;,
       colour = &quot;Age in\nmonths&quot;, 
       shape = &quot;Sex&quot;) +
  theme(aspect.ratio = 1)
  
pca_plot_GSE97113</code></pre>
<p><img src="figure/import-data.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><p>Overall there is not a huge difference (PC1 only accounts for 11.5% variation). However, the 4-month-olds separate out from the 1-2 month olds. This makes sense as they are quite a bit older. There is also some noticeable separation across PC2 of the male and female samples.</p></li>
<li><p>Dataset looks good, I will re-process the raw data for this one.</p></li>
</ul>
<pre class="bash"><code>split -l 1 ena download --additional-suffix=.txt
count=1; for file in *.txt; do mv $file  ${count}_${file}; count=$(expr $count + 1);  done</code></pre>
<pre><code>split: ena: No such file or directory
mv: rename *.txt to 1_*.txt: No such file or directory</code></pre>
</div>
<div id="frontal-cortex-6-and-11-months-3-biological-replicates-wt" class="section level2">
<h2>3. <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE142633">Frontal cortex, 6 and 11 months (3 biological replicates + WT)</a></h2>
<ul>
<li>The counts provided for this one has been estimated at the transcript level, although the authors have provided counts at gene level.</li>
</ul>
<pre class="r"><code>counts &lt;- read_tsv(here(&quot;data/datasets&quot;,
                        &quot;/mouse/GSE142633_RAW_transcripts.txt&quot;)) %&gt;%
  set_colnames(gsub(x=colnames(.), pattern =&quot;_.*$&quot;, replacement = &quot;&quot;))</code></pre>
<pre><code>Parsed with column specification:
cols(
  GSM4233218_WT_806_6M_FC.txt = col_double(),
  GSM4233219_WT_807_6M_FC.txt = col_double(),
  GSM4233220_WT_817_6M_FC.txt = col_double(),
  GSM4233221_WT_819_6M_FC.txt = col_double(),
  GSM4233222_TG_967_6M_FC.txt = col_double(),
  GSM4233223_TG_968_6M_FC.txt = col_double(),
  GSM4233224_TG_970_6M_FC.txt = col_double(),
  GSM4233225_WT_178_11M_FC.txt = col_double(),
  GSM4233226_WT_772_11M_FC.txt = col_double(),
  GSM4233227_WT_774_11M_FC.txt = col_double(),
  GSM4233228_TG_180_11M_FC.txt = col_double(),
  GSM4233229_TG_183_11M_FC.txt = col_double(),
  GSM4233230_TG_795_11M_FC.txt = col_double()
)</code></pre>
<pre class="r"><code>genes &lt;- read_tsv(here(&quot;data/datasets/mouse&quot;, 
                       &quot;GSM4233218_WT_806_6M_FC.genes.results.txt&quot;)) %&gt;%
  dplyr::select(gene_id, `transcript_id(s)`)</code></pre>
<pre><code>Parsed with column specification:
cols(
  gene_id = col_character(),
  `transcript_id(s)` = col_character(),
  length = col_double(),
  effective_length = col_double(),
  expected_count = col_double(),
  TPM = col_double(),
  FPKM = col_double()
)</code></pre>
<ul>
<li>Retrieve sample info</li>
</ul>
<pre class="r"><code>geo &lt;- getGEO(&quot;GSE142633&quot;, GSEMatrix = TRUE)</code></pre>
<pre><code>Found 1 file(s)</code></pre>
<pre><code>GSE142633_series_matrix.txt.gz</code></pre>
<pre><code>Parsed with column specification:
cols(
  ID_REF = col_character(),
  GSM4233218 = col_character(),
  GSM4233219 = col_character(),
  GSM4233220 = col_character(),
  GSM4233221 = col_character(),
  GSM4233222 = col_character(),
  GSM4233223 = col_character(),
  GSM4233224 = col_character(),
  GSM4233225 = col_character(),
  GSM4233226 = col_character(),
  GSM4233227 = col_character(),
  GSM4233228 = col_character(),
  GSM4233229 = col_character(),
  GSM4233230 = col_character()
)</code></pre>
<pre><code>File stored at: </code></pre>
<pre><code>/var/folders/mf/j7h9p8l55k9fb0rgm_f753cc0000gn/T//RtmpzchMFb/GPL24247.soft</code></pre>
<pre class="r"><code>samples &lt;- geo$GSE142633_series_matrix.txt.gz@phenoData %&gt;% 
  pData %&gt;%
  as_tibble %&gt;% 
  dplyr::select(title, description,
                geo_accession,
                starts_with(&quot;characteristics&quot;)) %&gt;%
  dplyr::rename(genotype = characteristics_ch1,
                tissue = characteristics_ch1.1,
                age = characteristics_ch1.2) %&gt;%
  dplyr::mutate(genotype = gsub(x=genotype, pattern=&quot;strain: &quot;, 
                                replacement = &quot;&quot;) %&gt;% as.factor,
                genotype = gsub(x=genotype, pattern=&quot;B6SJLF1/J&quot;,
                                replacement = &quot;WT&quot;),
                tissue = gsub(x=tissue, pattern=&quot;tissue: | of brain&quot;, 
                              replacement = &quot;&quot;) %&gt;% 
                  as.factor,
                age = gsub(x=age, pattern=&quot;age: | months&quot;, replacement = &quot;&quot;) %&gt;% 
                  as.factor,
                title = gsub(x=title,pattern=&quot; mRNAseq&quot;, replacement=&quot;&quot;),
                rep = c(1:4, 1:3, 1:3, 1:3),
                description = paste(genotype, age, rep, sep=&quot;_&quot;)) 

colnames(counts)==samples$geo_accession # All TRUE</code></pre>
<pre><code> [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code>colnames(counts) &lt;- samples$description
counts %&lt;&gt;% as.data.frame %&gt;% set_rownames(genes$gene_id)</code></pre>
<ul>
<li>Create DGEList to store counts, samples, and genes.</li>
</ul>
<pre class="r"><code>dge_GSE142633 &lt;- DGEList(counts=counts, 
                        samples = samples,
                        genes = genes, remove.zeros = TRUE)</code></pre>
<pre><code>Removing 24698 rows with all zero counts</code></pre>
<ul>
<li>Quick PCA to check samples and their similarity to each other.</li>
</ul>
<pre class="r"><code>pca &lt;- prcomp(t(cpm(dge_GSE142633, log = TRUE)))

shape_factor &lt;- paste0(samples$genotype, &quot;_&quot;, samples$age, &quot;_mths&quot;) %&gt;% 
  factor(levels = unique(.))

pca_plot_GSE142633 &lt;- pca$x %&gt;% magrittr::extract(, c(&quot;PC1&quot;, &quot;PC2&quot;)) %&gt;%
  set_colnames(c(&quot;PCa&quot;, &quot;PCb&quot;)) %&gt;%
  as.data.frame %&gt;%
  rownames_to_column(&quot;samples&quot;) %&gt;%
  left_join((dge_GSE142633$samples %&gt;% rownames_to_column(&quot;samples&quot;)), by=&quot;samples&quot;) %&gt;%
  ggplot(aes(x=PCa, y = PCb, shape = genotype, colour = as.factor(age))) +
  geom_point(shape = 1,size = 3.3)+
  geom_point(alpha = 1,size=3) + 
  scale_shape_manual(values = c(1,16)) + 
  #scale_shape_manual(values = c(0,15,1,16,2,17),
  #                   labels = levels(shape_factor)) +
  labs(x = &quot;Principal Component 1 (19.4%)&quot;, 
       y = &quot;Principal Component 2 (13.2%)&quot;,
       colour = &quot;Age in\nmonths&quot;, 
       shape = &quot;Genotype&quot;) +
  theme(aspect.ratio = 1)
  
pca_plot_GSE142633</code></pre>
<p><img src="figure/import-data.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>Samples separate cleanly by age and genotype. Age is the leading source of variation (PC1) while genotype is across PC2. Dataset is good to proceed, and I will download and re-process the raw data.</li>
</ul>
</div>
<div id="and-12-months-mouse-cortex" class="section level2">
<h2>4. 3,6 and 12 months, mouse cortex</h2>
<ul>
<li>This dataset was used in the IRE analysis paper. Here we import the sample table below.</li>
</ul>
<pre class="r"><code>samples &lt;- getGEO(filename = here::here(&quot;data&quot;, &quot;datasets&quot;, &quot;mouse&quot;,
                                        &quot;GSE140286_series_matrix.txt&quot;), 
                  getGPL = FALSE) %&gt;%
  phenoData %&gt;%
  pData %&gt;%
  as_tibble</code></pre>
<pre><code>Parsed with column specification:
cols(
  ID_REF = col_character(),
  GSM4158886 = col_character(),
  GSM4158887 = col_character(),
  GSM4158888 = col_character(),
  GSM4158889 = col_character(),
  GSM4158890 = col_character(),
  GSM4158891 = col_character(),
  GSM4158892 = col_character(),
  GSM4158893 = col_character(),
  GSM4158894 = col_character(),
  GSM4158895 = col_character(),
  GSM4158896 = col_character(),
  GSM4158897 = col_character(),
  GSM4158898 = col_character(),
  GSM4158899 = col_character(),
  GSM4158900 = col_character()
)</code></pre>
<ul>
<li>Keep relevant columns:</li>
</ul>
<pre class="r"><code>samples %&lt;&gt;% dplyr::select(description.1,
                          title, 
                          starts_with(&quot;age&quot;),
                          starts_with(&quot;genotype&quot;),
                          starts_with(&quot;tissue&quot;),
                          contains(&quot;Sample_characteristics_ch1&quot;))  %&gt;% 
    dplyr::rename(id = description.1,
                name = title,
                age = `age:ch1`,
                genotype = `genotype:ch1`,
                tissue = `tissue:ch1`) %&gt;%
  dplyr::mutate(age = gsub(x = age, 
                           pattern = &quot; months&quot;, 
                           replacement = &quot;&quot;),
                age = as.numeric(age)) %&gt;%
  dplyr::mutate(genotype = gsub(x = name, 
                                pattern = &quot;_(.)*$&quot;, 
                                replacement = &quot;&quot;),
                genotype = factor(genotype, 
                                  levels = c(&quot;5XFAD&quot;,&quot;WT&quot;)))

samples</code></pre>
<pre><code># A tibble: 15 x 5
   id               name           age genotype tissue
   &lt;fct&gt;            &lt;fct&gt;        &lt;dbl&gt; &lt;fct&gt;    &lt;chr&gt; 
 1 SJMMALZ058765_G1 WT_3mo_1         3 WT       Cortex
 2 SJMMALZ058766_G1 WT_3mo_2         3 WT       Cortex
 3 SJMMALZ058768_G1 5XFAD_3mo_1      3 5XFAD    Cortex
 4 SJMMALZ058769_G1 5XFAD_3mo_2      3 5XFAD    Cortex
 5 SJMMALZ058771_G1 WT_6mo_1         6 WT       Cortex
 6 SJMMALZ058772_G1 WT_6mo_2         6 WT       Cortex
 7 SJMMALZ058773_G1 WT_6mo_3         6 WT       Cortex
 8 SJMMALZ058774_G1 5XFAD_6mo_1      6 5XFAD    Cortex
 9 SJMMALZ058775_G1 5XFAD_6mo_2      6 5XFAD    Cortex
10 SJMMALZ058776_G1 5XFAD_6mo_3      6 5XFAD    Cortex
11 SJMMALZ058778_G1 WT_12mo_2       12 WT       Cortex
12 SJMMALZ058782_G1 WT_12mo_3       12 WT       Cortex
13 SJMMALZ058780_G1 5XFAD_12mo_1    12 5XFAD    Cortex
14 SJMMALZ058781_G1 5XFAD_12mo_2    12 5XFAD    Cortex
15 SJMMALZ058779_G1 5XFAD_12mo_3    12 5XFAD    Cortex</code></pre>
<ul>
<li>Import counts matrix:</li>
</ul>
<pre class="r"><code>counts &lt;- here(&quot;data&quot;, &quot;datasets&quot;, &quot;mouse&quot;,
               &quot;GSE140286_FAD_time_series_rnaseq_raw_count.txt&quot;) %&gt;%
  read_tsv</code></pre>
<pre><code>Parsed with column specification:
cols(
  Gene = col_character(),
  SJMMALZ058765_G1 = col_double(),
  SJMMALZ058766_G1 = col_double(),
  SJMMALZ058768_G1 = col_double(),
  SJMMALZ058769_G1 = col_double(),
  SJMMALZ058771_G1 = col_double(),
  SJMMALZ058772_G1 = col_double(),
  SJMMALZ058773_G1 = col_double(),
  SJMMALZ058774_G1 = col_double(),
  SJMMALZ058775_G1 = col_double(),
  SJMMALZ058776_G1 = col_double(),
  SJMMALZ058778_G1 = col_double(),
  SJMMALZ058779_G1 = col_double(),
  SJMMALZ058780_G1 = col_double(),
  SJMMALZ058781_G1 = col_double(),
  SJMMALZ058782_G1 = col_double()
)</code></pre>
<pre class="r"><code>counts</code></pre>
<pre><code># A tibble: 37,991 x 16
   Gene  SJMMALZ058765_G1 SJMMALZ058766_G1 SJMMALZ058768_G1 SJMMALZ058769_G1
   &lt;chr&gt;            &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;
 1 ENSM…            23145            23668            27445            24838
 2 ENSM…                0                0                0                1
 3 ENSM…                0                3                2                3
 4 ENSM…             6664             6084             8344             8552
 5 ENSM…                3                4                6                6
 6 ENSM…               73               11               63               24
 7 ENSM…            17525            18604            20293            22173
 8 ENSM…             1417             1840             1610             2683
 9 ENSM…            20938            20240            22835            26377
10 ENSM…              899             1282             1276             1702
# … with 37,981 more rows, and 11 more variables: SJMMALZ058771_G1 &lt;dbl&gt;,
#   SJMMALZ058772_G1 &lt;dbl&gt;, SJMMALZ058773_G1 &lt;dbl&gt;, SJMMALZ058774_G1 &lt;dbl&gt;,
#   SJMMALZ058775_G1 &lt;dbl&gt;, SJMMALZ058776_G1 &lt;dbl&gt;, SJMMALZ058778_G1 &lt;dbl&gt;,
#   SJMMALZ058779_G1 &lt;dbl&gt;, SJMMALZ058780_G1 &lt;dbl&gt;, SJMMALZ058781_G1 &lt;dbl&gt;,
#   SJMMALZ058782_G1 &lt;dbl&gt;</code></pre>
<pre class="r"><code>sampleNames &lt;- counts[,-c(1)] %&gt;% colnames
sampleNames</code></pre>
<pre><code> [1] &quot;SJMMALZ058765_G1&quot; &quot;SJMMALZ058766_G1&quot; &quot;SJMMALZ058768_G1&quot; &quot;SJMMALZ058769_G1&quot;
 [5] &quot;SJMMALZ058771_G1&quot; &quot;SJMMALZ058772_G1&quot; &quot;SJMMALZ058773_G1&quot; &quot;SJMMALZ058774_G1&quot;
 [9] &quot;SJMMALZ058775_G1&quot; &quot;SJMMALZ058776_G1&quot; &quot;SJMMALZ058778_G1&quot; &quot;SJMMALZ058779_G1&quot;
[13] &quot;SJMMALZ058780_G1&quot; &quot;SJMMALZ058781_G1&quot; &quot;SJMMALZ058782_G1&quot;</code></pre>
<ul>
<li>Check the “id” column in <code>samples</code> corresponds to the <code>sampleNames</code>.</li>
</ul>
<pre class="r"><code>samples$id %in% sampleNames # Are all samples in sample table present in counts samples?</code></pre>
<pre><code> [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code>samples$id == sampleNames # Is the order the same?</code></pre>
<pre><code> [1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE
[13]  TRUE  TRUE FALSE</code></pre>
<ul>
<li>The ordering of samples isn’t the same across the sample table and the counts matrix, so we will re-order the rows in <code>samples</code> so that they match the order of <code>sampleNames</code> from the <code>counts</code> matrix.</li>
</ul>
<pre class="r"><code>samples &lt;- samples %&gt;% 
  as.data.frame %&gt;%
  set_rownames(.$id) %&gt;%
  magrittr::extract(sampleNames, )</code></pre>
<pre class="r"><code>samples</code></pre>
<pre><code>                               id         name age genotype tissue
SJMMALZ058765_G1 SJMMALZ058765_G1     WT_3mo_1   3       WT Cortex
SJMMALZ058766_G1 SJMMALZ058766_G1     WT_3mo_2   3       WT Cortex
SJMMALZ058768_G1 SJMMALZ058768_G1  5XFAD_3mo_1   3    5XFAD Cortex
SJMMALZ058769_G1 SJMMALZ058769_G1  5XFAD_3mo_2   3    5XFAD Cortex
SJMMALZ058771_G1 SJMMALZ058771_G1     WT_6mo_1   6       WT Cortex
SJMMALZ058772_G1 SJMMALZ058772_G1     WT_6mo_2   6       WT Cortex
SJMMALZ058773_G1 SJMMALZ058773_G1     WT_6mo_3   6       WT Cortex
SJMMALZ058774_G1 SJMMALZ058774_G1  5XFAD_6mo_1   6    5XFAD Cortex
SJMMALZ058775_G1 SJMMALZ058775_G1  5XFAD_6mo_2   6    5XFAD Cortex
SJMMALZ058776_G1 SJMMALZ058776_G1  5XFAD_6mo_3   6    5XFAD Cortex
SJMMALZ058778_G1 SJMMALZ058778_G1    WT_12mo_2  12       WT Cortex
SJMMALZ058779_G1 SJMMALZ058779_G1 5XFAD_12mo_3  12    5XFAD Cortex
SJMMALZ058780_G1 SJMMALZ058780_G1 5XFAD_12mo_1  12    5XFAD Cortex
SJMMALZ058781_G1 SJMMALZ058781_G1 5XFAD_12mo_2  12    5XFAD Cortex
SJMMALZ058782_G1 SJMMALZ058782_G1    WT_12mo_3  12       WT Cortex</code></pre>
<pre class="r"><code>samples$id == sampleNames # Confirm order matches now</code></pre>
<pre><code> [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<ul>
<li>Extract gene IDs</li>
</ul>
<pre class="r"><code>genes &lt;- data.frame(ensembl_gene_id = counts$Gene)</code></pre>
<pre class="r"><code>counts %&lt;&gt;% column_to_rownames(&quot;Gene&quot;)</code></pre>
<pre class="r"><code>colnames(counts) &lt;- samples$name</code></pre>
<ul>
<li>Create DGEList</li>
</ul>
<pre class="r"><code>dge_GSE140286 &lt;- DGEList(
  counts = counts, 
  genes = genes,
  samples = samples,
  remove.zeros = TRUE
)</code></pre>
<pre><code>Removing 8336 rows with all zero counts</code></pre>
<ul>
<li>PCA</li>
</ul>
<pre class="r"><code>pca &lt;- prcomp(t(cpm(dge_GSE140286, log = TRUE)))

shape_factor &lt;- paste0(samples$genotype, &quot;_&quot;, samples$age) %&gt;% 
  factor(levels = unique(.))

pca_plot_GSE140286 &lt;- pca$x %&gt;% magrittr::extract(, c(&quot;PC1&quot;, &quot;PC2&quot;)) %&gt;%
  set_colnames(c(&quot;PCa&quot;, &quot;PCb&quot;)) %&gt;%
  as.data.frame %&gt;%
  rownames_to_column(&quot;samples&quot;) %&gt;%
  left_join((dge_GSE140286$samples %&gt;% rownames_to_column(&quot;samples&quot;)), by=&quot;samples&quot;) %&gt;%
  ggplot(aes(x=PCa, y = PCb, shape = genotype, colour = as.factor(age))) +
  geom_point(shape = 1,size = 3.3)+
  geom_point(alpha = 1,size=3) + 
  scale_shape_manual(values = c(1,16)) + 
  #scale_shape_manual(values = c(0,15,1,16,2,17),
  #                   labels = levels(shape_factor)) +
  labs(x = &quot;Principal Component 1 (20.5%)&quot;, 
       y = &quot;Principal Component 2 (11%)&quot;,
       colour = &quot;Age (Months)&quot;, 
       shape = &quot;Genotype&quot;) +
  theme(aspect.ratio = 1)
  
pca_plot_GSE140286</code></pre>
<p><img src="figure/import-data.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><p>In contrast to the other datasets above, the main source of variation in the data is not age but genotype.</p></li>
<li><p>Overall separation is decent, and I will re-process the data.</p></li>
</ul>
</div>
<div id="export-objects" class="section level2">
<h2>Export Objects</h2>
<ul>
<li>GSE97113 dataset:</li>
</ul>
<pre class="r"><code>dge_GSE97113 %&gt;% saveRDS(here(&quot;data/datasets/mouse/dge_GSE97113.rds&quot;))
pca_plot_GSE97113 %&gt;% export::graph2pdf(here(&quot;data/datasets/mouse/pca_plot_GSE97113&quot;))</code></pre>
<ul>
<li>GSE142633 dataset:</li>
</ul>
<pre class="r"><code>dge_GSE142633 %&gt;% saveRDS(here(&quot;data/datasets/mouse/dge_GSE142633.rds&quot;))
pca_plot_GSE142633 %&gt;% export::graph2pdf(here(&quot;data/datasets/mouse/pca_plot_GSE142633&quot;))</code></pre>
<ul>
<li>GSE140286 dataset:</li>
</ul>
<pre class="r"><code>dge_GSE140286 %&gt;% saveRDS(here(&quot;data/datasets/mouse/dge_GSE140286.rds&quot;))
pca_plot_GSE140286 %&gt;% export::graph2pdf(here(&quot;data/datasets/mouse/pca_plot_GSE140286&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.2 (2018-12-20)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplot2_3.2.1       UpSetR_1.4.0        edgeR_3.24.3       
 [4] limma_3.38.3        dplyr_0.8.3         readr_1.3.1        
 [7] tibble_2.1.3        GEOquery_2.50.5     Biobase_2.42.0     
[10] BiocGenerics_0.28.0 magrittr_1.5        here_0.1           

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5 locfit_1.5-9.1   xfun_0.11        purrr_0.3.3     
 [5] lattice_0.20-38  colorspace_1.4-1 vctrs_0.2.0      htmltools_0.4.0 
 [9] yaml_2.2.0       utf8_1.1.4       rlang_0.4.1      later_1.0.0     
[13] pillar_1.4.2     glue_1.3.1       withr_2.1.2      lifecycle_0.1.0 
[17] plyr_1.8.4       stringr_1.4.0    munsell_0.5.0    gtable_0.3.0    
[21] workflowr_1.6.1  evaluate_0.14    labeling_0.3     knitr_1.26      
[25] httpuv_1.5.2     curl_4.2         fansi_0.4.0      Rcpp_1.0.3      
[29] promises_1.1.0   backports_1.1.5  scales_1.0.0     fs_1.3.1        
[33] gridExtra_2.3    hms_0.5.2        digest_0.6.22    stringi_1.4.3   
[37] rprojroot_1.3-2  grid_3.5.2       cli_1.1.0        tools_3.5.2     
[41] lazyeval_0.2.2   crayon_1.3.4     tidyr_1.0.0      pkgconfig_2.0.3 
[45] zeallot_0.1.0    ellipsis_0.3.0   xml2_1.2.2       assertthat_0.2.1
[49] rmarkdown_1.17   R6_2.4.1         git2r_0.26.1     compiler_3.5.2  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
