---
title: "Clustering merged AD Data"
author: "Nhi Hin"
date: "2020-08-20"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
library(here)
library(magrittr)
library(ggplot2)
library(tibble)
library(dplyr)
library(edgeR)
library(reshape2)
library(pheatmap)

# ggplot2 theme
theme_set(theme_bw())

# Save pheatmap
save_pheatmap_pdf <- function(x, filename, width=8, height=11) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}
```


## Introduction

- Hypothesis is that using AES-PCA to summarise a gene counts matrix will 
result in a reduction of noise as well as greater emphasis of the 
biologically invariant and important signal in the data, compared to 
just the regular gene counts matrix.

## 1. Import Data

- We will be importing the fAD dataset by Antonell et al. (microarray, 
pre-processed), as well as the Mayo Clinic RNA-seq study containing 
mostly sAD patients (RNA-seq, pre-processed). 

- Details on the datasets:
    - **fAD dataset platform**: Affymetrix Human Gene 1.1 ST Array [transcript (gene) version]
    - **sAD dataset platform**: Total RNA-seq, 100bp reads

```{r}
# Mayo Dataset
mayoDGE <- readRDS(here("data", "datasets", "mayo", "dge.rds"))
# Extract relevant samples to correspond with those that 
# AES-PCA was run on:
mayoDGE$samples %<>% dplyr::filter(Diagnosis %in% c("AD", "Control"),
                                  Tissue == "TemporalCortex") 
mayoDGE$counts %<>% magrittr::extract(, mayoDGE$samples$SampleID)
dim(mayoDGE$counts)
dim(mayoDGE$samples)

# Antonell dataset
fADData <- readRDS(here("data", "datasets", "fad", "procData.rds"))
fADData$annot <- readRDS(here("data", "datasets", "fad", "annot.rds"))
# Get the probes into gene IDs.
fADData$annot %>% head
fADData$exprs2 <- fADData$exprs %>% rownames_to_column("PROBEID") %>%
  left_join(fADData$annot[, c("PROBEID", "ENSEMBL")], by = "PROBEID") %>%
  dplyr::filter(!is.na(ENSEMBL)) %>%
  dplyr::distinct(ENSEMBL, .keep_all = TRUE) %>%
  dplyr::select(-PROBEID) %>%
  column_to_rownames("ENSEMBL")
```

- PC1 values calculated using AES-PCA in the respective analyses.

```{r}
# # Antonell et al. dataset
# samples_fAD <- readRDS(here("data", "datasets", "fad", "samples.rds"))
# pc1_fAD <- readRDS(here("data", "datasets", 
#                                 "fad", "PC1_allSamples.rds"))
# 
# # Mayo Clinic RNA-seq (sporadic) AD dataset
# samples_mayo <- readRDS(here("data", "datasets", "mayo", 
#                                  "heatmapAnnot.rds"))
# pc1_mayo <- readRDS(here("data", "datasets", "mayo", 
#                                "PC1_TCsamples.rds"))

combinedAnnot_human <- readRDS(here("data", "PC1", "combinedAnnot.rds")) %>%
  rownames_to_column("Sample") %>%
  dplyr::select(Sample, Cluster, Diagnosis) %>%
  dplyr::rename(Group = Diagnosis)

combinedPC1_human <- readRDS(here("data", "PC1", "combinedPC1.rds"))
```

## 2. Merge Data

### 2.1. Gene expression Data

- In order to perform clustering on the datasets together we will need 
to merge them. 

- First the normalised gene expression data. 

```{r}
mergedDat <- mayoDGE %>% cpm(log=TRUE) %>%
  as.data.frame %>%
  rownames_to_column("ENSEMBL") %>%
  inner_join(fADData$exprs2 %>% rownames_to_column("ENSEMBL"), 
            by = "ENSEMBL") %>%
  column_to_rownames("ENSEMBL")

```

- Quick PCA to see how they look

```{r}
pca_mergedDat <- mergedDat %>% t %>% prcomp(scale.=TRUE)
summary(pca_mergedDat) # PC1: 75.9%, PC2: 8.1%

mergedPCA_Plot <- pca_mergedDat$x%>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  dplyr::select(Sample, PC1, PC2) %>%
  mutate(dataset = c(rep("sAD,\nRNA-seq",nrow(mayoDGE$samples)),
                     rep("eoAD,\nmicroarray",21)))%>%
  left_join(combinedAnnot_human[, c("Sample", "Group")], 
            by = "Sample") %>%
  dplyr::mutate(Diagnosis = case_when(
    Group == "eoAD" ~ "AD",
    Group == "eoAD_psen1" ~ "AD",
    Group == "control" ~ "control",
    Group == "AD" ~ "AD"
  ))  %>%
  ggplot2::ggplot(aes(x = PC1, y = PC2, 
                      colour = Diagnosis,
                      shape = dataset)) +
  geom_point(alpha = 0.6, size = 2.75) +
  scale_shape_manual(values = c(15,0))+
  scale_color_manual(values = c("red", "#777777"))+
  labs(x = "PC1 (75.9%)",
       y = "PC2 (8.1%)",
       shape = "Dataset /\nPlatform",
      color = "Diagnosis")+
  theme(aspect.ratio = 1) 
mergedPCA_Plot

mergedPCA_Plot %>% export::graph2pdf(here("data", "merged_human", "fig",
                                          "mergedPCA2.pdf"), width=6,height=6)
```

- As expected the dataset forms the main batch effect, 
accounting for 75.9% variation across PC1. 

### 2.2. Sparse PCs

- We will do the same basic PCA plot but instead of the gene expression 
data we will use the Sparse PCs calculated by AES-PCA. 

```{r}
pca_mergedPC <- combinedPC1_human  %>% prcomp()
summary(pca_mergedPC) #PC1 77.1%, PC2 8.6%

mergedPCA_Plot2 <- pca_mergedPC$x %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  dplyr::select(Sample, PC1, PC2) %>%
  left_join(combinedAnnot_human, by ="Sample") %>% 
  mutate(dataset = case_when(
    Cluster == 6 ~ "fAD",
    Cluster != 6 ~ "Mayo"
  )) %>% 
  mutate(Group = gsub(x = Group, 
                      pattern = "eoAD_psen1|eoAD",
                      replacement = "AD")) %>%
  mutate(dataset = gsub(x = dataset, pattern = "fAD", 
                        replacement = "eoAD,\nmicroarray")) %>%
  mutate(dataset = gsub(x = dataset, pattern = "Mayo",
                        replacement = "sAD, \nRNA-seq")) %>% 
  ggplot(aes(x = PC1, y = PC2, colour = Group, shape = dataset)) +
  geom_point(size = 2.5, alpha = 0.6) +
  scale_shape_manual(values = c(15,0, 10,16)) +
  scale_color_manual(values = c("red", "#777777")) + 
  theme(aspect.ratio = 1) +
  labs(x = "PC1 (77.1%)", y = "PC2 (8.6%)", 
       shape = "Dataset / \nPlatform",
      color  = "Diagnosis")

mergedPCA_Plot2

mergedPCA_Plot2 %>% export::graph2pdf(here("data", "merged_human",
                                           "fig", "merged_AESPCA2.pdf"),
                                      width=6, height=6)
```

- Using AES-PCA, the effect of the dataset/platform (batch effect) is gone. 
The samples now separate by their diagnosis. Overall, eoAD samples group 
similarly to AD samples. 
- However, there is a lot of variation in the sAD samples, particularly 
amongst both the AD and controls. This would make sense considering 
the varied ages that the dataset contains, meaning that we 
would expect quite varied gene expression patterns. 
- We can still see the overall trend of AD and control 
across PC2 though. 

## 2. Clustering of merged normalised data

- We will perform hierarchical clustering on the merged datasets 
before and after AES-PCA to further demonstrate this effect. 

- Clustering can be shown on a heatmap with dendrogram indicating the 
clusters formed from hierarchical clustering. 

### 2.1. Gene expression Data

- Note the heatmap takes a while to plot as it's using all the genes. 

```{r fig.height = 11}
# Create the sample annotation data.frame for the heatmap
heatmap_annot <- combinedAnnot_human %>% 
  mutate(Dataset = case_when(
    Cluster == 6 ~ "eoAD microarray",
    Cluster != 6 ~ "Mayo RNA-seq"
  ))%>%
  dplyr::select(-Cluster) %>%
  dplyr::mutate(Diagnosis = case_when(
    Group == "eoAD" ~ "AD",
    Group == "eoAD_psen1" ~ "AD",
    Group == "control" ~ "control",
    Group == "AD" ~ "AD"
  )) %>%
  dplyr::select(-Group) %>%
  column_to_rownames("Sample")
  
# Specify colours for the sample annotation bars on the heatmap
annot_colours<- list(Diagnosis = c(AD = "red", control = "#dddddd"),
                     Dataset = c("eoAD microarray" = "orange", 
                                 "Mayo RNA-seq" = "turquoise"))

# Heatmap overall palette, from blue-green-yellow-orange-red
palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

# Plot the heatmap
heatmap_mergedAD <- mergedDat %>%
  # rownames_to_column("Gene") %>%
  # dplyr::slice(1:1000) %>%
  # column_to_rownames("Gene") %>%
  pheatmap::pheatmap(
    cluster_rows = TRUE,
    annotation_colors = annot_colours,
    clustering_method = "average",
    cluster_cols = TRUE,
    scale = "row",
    annotation_col = heatmap_annot,
    color = palette
  )
```

- This heatmap clearly shows that the samples cluster mainly by dataset first. 
- Within the eoAD dataset, there is clear clustering into AD and control. 
- However, the sAD (Mayo) dataset is extremely messy reflecting the 
high heterogeneity. 

- Save the heatmap as a pdf. 

```{r eval=FALSE}
save_pheatmap_pdf(heatmap_mergedAD, 
                  here("data", "PC1", "fig", "mergedHeatmap.pdf"),
                  width = 8, height = 25)
```

### 2.2. Sparse PCs

- Plot the heatmap but using the sparse PCs. 

- We need to import the cluster information (see `pathwayPCA-sAD.Rmd`). 

- This is 3 clusters all with Jaccard index > 0.8 indicating high 
stability. 

```{r}
clustersPC1 <- readRDS(here("data", "datasets", "mayo", 
                             "km_clusters3_PC1.rds"))
```

- Append clusters to sample annotation table. 

```{r}
combinedAnnot_human %<>% dplyr::select(-Cluster)
combinedAnnot_human %<>% left_join(clustersPC1, by = "Sample")
combinedAnnot_human$Cluster[1:21] <- c(rep(4, 21))

```

- Plot heatmap

```{r}
heatmapAnnot_PCs <- combinedAnnot_human %>%
  mutate(Dataset = c(rep("eoAD microarray",21),
                     rep("Mayo RNA-seq", 132)))%>%
  dplyr::mutate(Diagnosis = case_when(
    Group == "eoAD" ~ "AD",
    Group == "eoAD_psen1" ~ "AD",
    Group == "control" ~ "control",
    Group == "AD" ~ "AD"
  ))  %>%
  dplyr::select(-Group)   %>%
  dplyr::mutate(Cluster = as.character(Cluster)) %>%
  column_to_rownames("Sample") 
  

annot_colours_PCs <- list(Diagnosis = c(AD = "red", control = "#dddddd"),
                          Dataset = c("eoAD microarray" = "blue",
                                      "Mayo RNA-seq" = "pink"),
                          Cluster = c("1" = "orange",
                                      "2"= "yellow",
                                      "3" = "turquoise",
                                      "4" = "white"))

# Plot the heatmap
heatmap_PCs <- combinedPC1_human %>% 
  as.data.frame %>% t %>%
  pheatmap::pheatmap(
    cluster_rows = TRUE,
    annotation_colors = annot_colours_PCs,
    clustering_method = "complete",
    cluster_cols = TRUE,
    scale = "row",
    annotation_col = heatmapAnnot_PCs,
    color = palette
  )
```

- It's still really messy. Maybe the tree was not such a good idea for 
a visualisation. Instead, I think I will stick with the PCA and show the 
idea that way where it's more clear. 
