---
title: "Mouse Datasets - Import Kallisto Data"
author: "Nhi Hin"
date: "2020-05-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r load-packages, include=FALSE}
library(AnnotationHub)
library(dplyr)
library(magrittr)
library(edgeR)
library(limma)
library(sleuth)
library(here)
library(readr)
library(tibble)
library(grid)
library(export)
library(reshape2)

# ggplot2 theme
theme_set(theme_bw())
```

## 0. Introduction

- The aim of this RMarkdown is to import the output *kallisto* 
transcript-level count estimates, summarise these to gene-level, perform 
filtering, and prepare them for differential gene expression analysis 
using *limma*. 


## 1. Pre-processing

- All raw data (fastq files) was downloaded to Phoenix from ENA. 

- After quality trimming with *AdapterRemover*, transcript expression was 
estimated using *kallisto* (v.0.45). 

- Default *kallisto* settings were used and the index was built from 
Ensembl release 98 *Mus musculus* cDNA (primary assembly only). 


## 2. Import Kallisto Output

- The output directories from running *kallisto* were imported using the 
`catchKallisto` function from *edgeR*. 

### 2.1. Dataset 1 (GSE97113)

```{r eval=FALSE}
# Get paths to the Results directories on Phoenix generated by Kallisto.
kallistoResults1 <- file.path("/Volumes/fastdir/datasets/mouse_1",
                              "3_quantData/kallisto") %>% 
  file.path %>%
  list.dirs
kallistoResults1 <- kallistoResults1[-1] 

# Takes a while to run so save out the object. 
importedKallisto1 <- catchKallisto(kallistoResults1) 
importedKallisto1 %>% saveRDS(here("data", "mouse", "kallisto",
                                   "importedKallisto1.rds"))
```
```{r include=FALSE}
importedKallisto1 <- readRDS(here("data", "datasets", "mouse", "kallisto",
                                   "importedKallisto1.rds"))
```

- Neaten column names:

```{r}
colnames(importedKallisto1$counts) %<>% 
  basename %>%
  gsub(x = ., pattern = "[[:digit:]]{1,2}_(S.*)_t", 
       replacement = "\\1")
```


### 2.2. Dataset 2 (GSE142633)

```{r eval=FALSE}
# Get paths to the Results directories on Phoenix generated by Kallisto.
kallistoResults2 <- file.path("/Volumes/fastdir/datasets/mouse_2",
                              "3_quantData/kallisto") %>% 
  file.path %>%
  list.dirs
kallistoResults2 <- kallistoResults2[-1] 

# Takes a while to run so save out the object. 
importedKallisto2 <- catchKallisto(kallistoResults2) 
importedKallisto2 %>% saveRDS(here("data", "mouse", "kallisto",
                                   "importedKallisto2.rds"))
```
```{r include=FALSE}
importedKallisto2 <- readRDS(here("data", "datasets","mouse", "kallisto",
                                   "importedKallisto2.rds"))
```

- Neaten column names:

```{r}
colnames(importedKallisto2$counts) %<>%
  basename %>%
  gsub(x = ., pattern = ".*(SRR.*$)", 
       replacement = "\\1")
```


### 2.2. Dataset 3 (GSE140286)

```{r eval=FALSE}
# Get paths to the Results directories on Phoenix generated by Kallisto.
kallistoResults3 <- file.path("/Volumes/fastdir/datasets/mouse_2",
                              "3_quantData/kallisto") %>% 
  file.path %>%
  list.dirs
kallistoResults3 <- kallistoResults3[-1] 

# Takes a while to run so save out the object. 
importedKallisto3 <- catchKallisto(kallistoResults3) 
importedKallisto3 %>% saveRDS(here("data", "mouse", "kallisto",
                                   "importedKallisto3.rds"))
```
```{r include=FALSE}
importedKallisto3 <- readRDS(here("data", "datasets", "mouse", "kallisto",
                                   "importedKallisto3.rds"))
```



## 3. Load Sample Tables

- Load the sample table for Dataset 1

```{r}
samples1 <- readRDS(here("data/datasets/mouse/dge_GSE97113.rds"))$samples
samples1 %<>% as.data.frame %>%
  mutate(group = paste0(genotype, sex, age, "mth"),
                    name = paste(group, "_",rep(1:5))) 

```

- Load the sample table for Dataset 2

```{r}
samples2 <- readRDS(here("data/datasets/mouse/dge_GSE142633.rds"))$samples

samples3 <-  readRDS(here("data/datasets/mouse/dge_GSE140286.rds"))$samples
```


## 4. Transcript-level counts

- Create matrix of the transcript-level count estimates with new names 
from the samples table:

```{r}
counts_t1 <- importedKallisto1$counts[, samples1$geo_accession] %>% 
  set_colnames(samples1$name)

# Preview
counts_t1 %>% as.data.frame %>% 
  rownames_to_column("transcript_id") %>% 
  as_tibble
```


## 5. Gene-Transcript Annotation

- To summarise to gene-level counts, we first need to prepare a `data.frame` 
that maps from Ensembl transcript to gene IDs. 

```{r}
# Search for and load relevant AnnotationHub object:
ah <- AnnotationHub()
ah %>%
  subset(grepl("rerio", species)) %>%
  subset(rdataclass == "EnsDb")
ensDb <- ah[["AH64906"]]

# Get gene and transcript information:
genes <- genes(ensDb)
transcripts <- transcripts(ensDb)

mapTx2Gene <- transcripts %>% 
  as.data.frame %>%
  dplyr::select(tx_id_version, gene_id)

# Preview
mapTx2Gene %>% 
  rownames_to_column("transcript_id")%>%
  as_tibble()
```

## 6. Gene-level counts

- Now we can summarise the transcript-level count estimates to gene-level 
count estimates by summing together the transcript-level counts corresponding 
to the same gene. We will also use the new names from the `samples` table. 

```{r}
counts_g <- counts_t %>%
  melt %>% 
  set_colnames(c("tx_id_version", "sample", "count")) %>% 
  left_join(mapTx2Gene) %>% 
  dcast(gene_id ~ sample, value.var = "count", fun.aggregate=sum)%>%
  magrittr::extract(,c("gene_id",sample_to_cov$sample)) 

# Preview
counts_g %>% as_tibble
```

## 7. Prepare DGEList

- We will store gene and transcript count estimates as DGEList 
objects for differential gene/transcript analysis later. 
DGEList objects also contain gene/transcript annotation and sample annotation 
information. 

- The DGEList for gene-level counts was created as follows: 

```{r}
counts_for_dge <- counts_g %>% 
  column_to_rownames("gene_id") %>%
  as.matrix

genes_for_dge <- data.frame(ensembl_gene_id = rownames(counts_for_dge)) %>%
  left_join(as.data.frame(genes) %>% 
              dplyr::select(gene_id, 
                            gene_name, 
                            gene_biotype, 
                            entrezid, 
                            description), 
            by = c("ensembl_gene_id"="gene_id"))

samples_for_dge <- sample_to_cov %>% set_rownames(.$sample)

dge <- DGEList(counts = counts_for_dge, 
               genes = genes_for_dge, 
               samples = samples_for_dge, 
               remove.zeros = TRUE) %>%  
  calcNormFactors("TMM")

dge
```

- The DGEList for transcript-level counts was created as follows:

```{r }
counts_for_dge_t <- counts_t %>% 
  as.data.frame %>% 
  rownames_to_column("ensembl_transcript_id") %>%
  magrittr::extract(, c("ensembl_transcript_id", sample_to_cov$sample)) %>%
  column_to_rownames("ensembl_transcript_id") %>% as.matrix

transcripts_for_dge_t <- data.frame(ensembl_transcript_id = 
                                      rownames(counts_t)) %>%
  left_join(mapTx2Gene, by = c("ensembl_transcript_id"="tx_id_version")) %>%
  left_join(as.data.frame(genes) %>% dplyr::select(gene_id, 
                                                   gene_name, 
                                                   gene_biotype, 
                                                   entrezid, 
                                                   description), 
            by = c("gene_id"))%>%
  dplyr::rename(ensembl_gene_id = gene_id)

dge_t <- DGEList(counts = counts_for_dge_t,
                 genes = transcripts_for_dge_t,
                 samples = samples_for_dge, remove.zeros = TRUE) %>%
  calcNormFactors("TMM")

dge_t
```


## 8. Initial PCA Visualisation

- We will use Principal Component Analysis to get an idea of the similarity 
of samples / libraries based on their gene and transcript expression. 

### 8.1. Gene level PCA

- The PCA below indicates that **age** is the largest source of variation in 
the data (across **PC1**). 

- There is some separation of samples due to **genotype** and **hypoxia** 
across **PC2**. 

```{r}
pca_analysis1 <- prcomp(t(cpm(dge, log=TRUE)))

pca_geneExpression <- pca_analysis1$x %>% 
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("samples") %>%
  left_join((dge$samples %>% rownames_to_column("samples")), by="samples") %>%
  ggplot(aes(x=PC1, y = PC2, 
             colour = Hypoxia, 
             shape = paste0(Genotype, "_", Age))) +
  geom_point(alpha = 0.7,size=3) + 
  scale_shape_manual(values = c(0,1,15,16),
                     labels = c("Q96K97/+, 24 months",
                                "Q96K97/+, 6 months", 
                                "+/+, 24 months", 
                                "+/+, 6 months")) +
  theme(aspect.ratio = 1) +
  scale_colour_manual(values = c("red","cornflowerblue"), 
                      labels = c("Hypoxia","Normoxia")) +
    labs(x = "Principal Component 1 (19.15%)",
         y = "Principal Component 2 (6.0%)", 
         colour = "Oxygen Level", shape = "Genotype and Age")

summary(pca_analysis1)
screeplot(pca_analysis1)
pca_geneExpression

export::graph2pdf(pca_geneExpression,
                  here("output", "zebrafish", "pca_geneExpression"))
```


### 8.2. Transcript level PCA

- At the transcript-level, the overall separation patterns look similar. 

```{r}
pca_analysis2 <- prcomp(t(cpm(dge_t, log=TRUE)))

summary(pca_analysis2)

pca_geneExpression2 <- pca_analysis2$x %>% 
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("samples") %>%
  left_join((dge$samples %>% rownames_to_column("samples")), 
            by="samples") %>%
  ggplot(aes(x=PC1, y = PC2, colour = Hypoxia, 
             shape = paste0(Genotype, "_", Age))) +
  geom_point(alpha = 0.7,size=3) + 
  scale_shape_manual(values = c(0,1,15,16),
                     labels = c("Q96K97/+, 24 months",
                                "Q96K97/+, 6 months", 
                                "+/+, 24 months", 
                                "+/+, 6 months")) +
  theme(aspect.ratio = 1) +
  scale_colour_manual(values = c("red","cornflowerblue"), 
                      labels = c("Hypoxia","Normoxia")) +
    labs(x = "Principal Component 1 (9.4%)", 
         y = "Principal Component 2 (7.8%)", 
         colour = "Oxygen Level", shape = "Genotype and Age")

summary(pca_analysis2)
screeplot(pca_analysis2)
pca_geneExpression2

export::graph2pdf(pca_geneExpression2,
                  here("output", "zebrafish", "pca_transcriptExpression"))
```

## 9. Filtering

### 9.1. Gene-level filtering

- We chose the threshold of at least 1 cpm in at least 4 samples 
based on the following density plots. 

```{r}
keepTheseGenes <- (rowSums(cpm(dge) > 1) >= 4) 

A <- dge %>% 
  cpm(log = TRUE) %>% 
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("A. Before filtering") +
  labs(x = "logCPM", y = "Density")

B <- dge %>% 
  cpm(log = TRUE) %>% 
  magrittr::extract(keepTheseGenes,) %>%
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("B. After filtering")+
  labs(x = "logCPM", y = "Density")

grid.newpage()
vp1 <- viewport(x = 0, y = 0, width = 0.5, height = 1, just = c(0, 0))
vp2 <- viewport(x = 0.5, y = 0, width = 0.5, height = 1, just = c(0,0))
print(A, vp = vp1)
print(B, vp  = vp2)
```

- Using this threshold, we filtered out `r table(keepTheseGenes)[[1]]`
genes from the original `r length(keepTheseGenes)` genes, giving the 
remaining `r table(keepTheseGenes)[[2]]` to be used in the analysis. 

```{r}
dge_filtered <- dge[keepTheseGenes,,keep.lib.sizes = FALSE] 
```


### 8.2. Transcript level filtering

- The threshold for filtering (at least 1 cpm in at least 4 samples) 
was chosen to be consistent with the gene level filtering.

```{r}
keepTheseTx <- (rowSums(cpm(dge_t) > 1) >= 4) 

A <- dge_t %>% 
  cpm(log = TRUE) %>% 
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("A. Before filtering") +
  labs(x = "logCPM", y = "Density")

B <- dge_t %>% 
  cpm(log = TRUE) %>% 
  magrittr::extract(keepTheseTx,) %>%
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("B. After filtering")+
  labs(x = "logCPM", y = "Density")

grid.newpage()
vp1 <- viewport(x = 0, y = 0, width = 0.5, height = 1, just = c(0, 0))
vp2 <- viewport(x = 0.5, y = 0, width = 0.5, height = 1, just = c(0,0))
print(A, vp = vp1)
print(B, vp  = vp2)
```

- Using this threshold, we filtered out `r table(keepTheseTx)[[1]]` 
transcripts from the original `r length(keepTheseTx)` transcripts, 
giving the remaining `r table(keepTheseTx)[[2]]` to be used in the analysis. 

```{r}
dge_t_filtered <- dge_t[keepTheseTx,,keep.lib.sizes = FALSE] 
```



## Export objects

- Gene-level and transcript-level counts

```{r eval=FALSE}
counts_t %>% saveRDS(here("data", "zebrafish", "counts_t.rds"))
counts_g %>% saveRDS(here("data", "zebrafish", "counts_g.rds"))

counts_t %>% 
  as.data.frame() %>% 
  rownames_to_column("ensembl_transcript_id") %>% 
  write_tsv(here("data", "zebrafish", "counts_t.txt"))

counts_g %>% 
  as.data.frame() %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  write_tsv(here("data", "zebrafish", "counts_g.txt"))
```

- `DGEList` objects

```{r eval=FALSE}
dge %>% saveRDS(here("data", "zebrafish", "dge_g.rds"))
dge_t %>% saveRDS(here("data", "zebrafish", "dge_t.rds"))
```

- Filtered `DGEList` objects

```{r eval=FALSE}
dge_filtered %>% saveRDS(here("data", "zebrafish", "dge_g_filtered.rds"))
dge_t_filtered %>% saveRDS(here("data", "zebrafish", "dge_t_filtered.rds"))
```

