---
title: "Testing MultiPLIER"
author: "Nhi Hin"
date: "2020-05-20"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r include=FALSE}
library(PLIER)
library(here)
library(AnnotationHub)
library(biomaRt)
library(org.Hs.eg.db)
library(magrittr)
library(dplyr)
library(edgeR)
library(pheatmap)
library(tibble)
library(ggplot2)
theme_set(theme_bw())
```


## Introduction

- MultiPLIER is a PLIER model trained on the entire *recount2* dataset. 
*recount2* contains many different RNA-seq human datasets from multiple 
tissues, cell types, and diseases. 

- The resulting MultiPLIER model supposedly contains surrogate variables 
(SVs) that are sensitive compared to a PLIER model trained on one dataset. 

From the [MultiPLIER paper](https://www.sciencedirect.com/science/article/pii/S240547121930119X#bib25):

<blockquote>
In summary, MultiPLIERâ€™s features have biological relevance 
and can be directly compared across datasets making it 
particularly valuable for and a conceptual advance in 
the integrative analysis of complex human disease.
</blockquote>

and

<blockquote>
Most gene expression datasets generated by individual 
researchers are too small to fully benefit
from unsupervised machine-learning methods. In
the case of rare diseases, there may be too few cases
available, even when multiple studies are combined.
To address this challenge, we utilize transfer learning
to extract coordinated expression patterns and
use learned patterns to analyze small rare disease
datasets.
</blockquote>

- In order to apply the MultiPLIER model to a dataset, we need to 
do the following:

    1. Convert gene IDs to human gene symbols
    2. Provide the gene count matrix to MultiPLIER's 
    `GetNewDataB()` function.
    

## 1. Zebrafish Dataset

- Import in the DGEList object previously created:

```{r}
dge_zeb <- readRDS(here("data", "datasets", "zebrafish", "dge.rds"))
dge_zeb
```

- Get zebrafish gene information

```{r}
ah <- AnnotationHub()
# ah %>%
#   subset(grepl("rerio", species)) %>%
#   subset(rdataclass == "EnsDb")
ensDb <- ah[["AH69169"]] # Ensembl 96 EnsDb for Danio rerio

# Get gene info for zebrafish
genes <- genes(ensDb)
genes %<>% 
  as.data.frame %>% 
  dplyr::distinct(gene_id, .keep_all=TRUE)
```

- We need human gene symbols so the zebrafish IDs will be converted using 
biomart. 

```{r}
# Sometimes biomart doesn't connect so I've saved the mart object out
# zebMart <- useMart("ensembl", "drerio_gene_ensembl")
zebMart <- readRDS(here("data", "datasets","zebrafish", "zebMart.rds"))
getFromBiomart <- c("ensembl_gene_id", "hsapiens_homolog_ensembl_gene")
zebAndHumanEnsGenes <- getBM(getFromBiomart, 
                             values = unique(genes$gene_id), 
                             mart = zebMart) %>%
  set_colnames(c("zebrafish_ensembl", "human_ensembl")) %>%
  # Only keep zebrafish genes which have human homolog
  dplyr::filter(human_ensembl!="") 
```

- Convert human ensembl to symbol

```{r}
ah %>%
  subset(grepl("sapiens", species)) %>%
  subset(rdataclass == "EnsDb")
humanEnsDb <- ah[["AH69187"]] # Ensembl 96 EnsDb for Homo sapiens
human_genes <- genes(humanEnsDb) %>% 
  as.data.frame %>%
  dplyr::distinct(gene_id,.keep_all=TRUE)

humanEnsemblAndSymbol <- human_genes %>%
  dplyr::select(gene_id, symbol)

zebEns2humanSym <- zebAndHumanEnsGenes %>% 
  left_join(humanEnsemblAndSymbol, by = c("human_ensembl"="gene_id")) 
```

- Get the counts matrix:

```{r}
counts_zeb <- dge_zeb %>%cpm(log=TRUE)%>%
  as.data.frame %>%
  rownames_to_column("zebrafish_ensembl") %>%
  left_join(zebEns2humanSym, by = "zebrafish_ensembl") %>%
  dplyr::distinct(symbol, .keep_all=TRUE) %>%
  dplyr::filter(!is.na(symbol)) %>%
  dplyr::filter(symbol!="")%>%
  dplyr::select(-human_ensembl, -zebrafish_ensembl) %>%
  column_to_rownames("symbol") %>%
  as.matrix
```

- Save it out and upload the rds to Phoenix HPC

```{r eval=FALSE}
saveRDS(counts_zeb, here("data", "multiplier", "zeb_matrix2.rds"))
```

- Previously downloaded the MultiPLIER files from 
[Figshare](https://figshare.com/articles/MultiPLIER_Fileset/6982919/2). 
This had to be submitted as a Phoenix job as the files were 82 Gb in 
total.

```{r engine="bash", eval=FALSE}
wget --no-check-certificate https://figshare.com/articles/MultiPLIER_Fileset/6982919/2
```

- The following code was run on Phoenix as I'm running low on internet data and 
couldn't download the MultiPLIER model locally.

```{r eval=FALSE}
library(PLIER)
model  <- readRDS("/fast/users/a1669135/multi-plier/data/recount2_PLIER_data/recount_PLIER_model.RDS")
source(file.path("/fast/users/a1669135/multi-plier/util", "plier_util.R"))
zebMatrix <- readRDS("/fast/users/a1669135/zeb_matrix.rds")
recountRes <- GetNewDataB(zebMatrix, model)
saveRDS(recountRes, "/fast/users/a1669135/recountRes.rds")
```

### 1.1. Import in MultiPLIER results

```{r}
recountRes <- readRDS(here("data", "multiplier", "recountRes2.rds"))
dim(recountRes)
```

- There are 987 features in the MultiPLIER model

### 1.2. Heatmap of latent variables

```{r}
idx_relevantLV <- c(grep("SVM", rownames(recountRes)))
rownames(recountRes) <- abbreviate(rownames(recountRes))

# Heatmap of all LVs and samples
recountRes %>% pheatmap::pheatmap(scale="row",
                                  cluster_cols = FALSE)

# Subset of some 6 month old samples (WT normoxia, WT hypoxia, mutant normoxia)
# and a subset of the immune cell type LVs
recountRes[idx_relevantLV, c(1:4,5:8,9:12)] %>%
  pheatmap::pheatmap(scale="row", cluster_cols = FALSE)
```

- No matter how I look at it, it doesn't seem like these features 
can distinguish between sample groups at all...

### 1.3. PCA

- Last test using PCA on the latent variables.

```{r}
pca <- recountRes %>% t %>% prcomp()
summary(pca) # PC1: 20%, PC2: 15%

pca_plot <- pca$x %>%
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame() %>%
  rownames_to_column("samples") %>%
  left_join((dge_zeb$samples %>% rownames_to_column("samples")), by="samples") %>%
  ggplot(aes(x=PC1, y = PC2, 
             colour = as.factor(paste0(Age, "mths, ", Genotype)), 
             shape = Gender)) +
  geom_point(alpha = 0.7,size=3) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (20%)",
         y = "Principal Component 2 (15%)", 
         colour = "Age in \nmonths and \ngenotype", shape = "Sex") 

pca_plot
```

- For comparison, a PCA of the original gene expression counts (logCPM):

```{r}
pca_orig <- dge_zeb %>% cpm(log=TRUE) %>% t %>% prcomp()
summary(pca_orig) # PC1: 19.2%, PC2: 6%

pca_orig_plot <- pca_orig$x %>%
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame() %>%
  rownames_to_column("samples") %>%
  left_join((dge_zeb$samples %>% rownames_to_column("samples")), by="samples") %>%
  ggplot(aes(x=PC1, y = PC2, 
             colour = as.factor(paste0(Age, "mths, ", Genotype)), 
             shape = Gender)) +
  geom_point(alpha = 0.7,size=3) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (19.2%)",
         y = "Principal Component 2 (6%)", 
         colour = "Age in \nmonths and \ngenotype", shape = "Sex") 

pca_orig_plot
```

- Surprisingly, despite only having 987 features (latent variables) in the 
MultiPLIER model, the PCA somehow even looks better than the one based on 
~25,000 genes. Weirdly enough the mutant samples which looked kind of like 
outliers in the plot with all genes group back with their group in the 
LV PCA. 

- Separation by genotype isn't 100% across PC1. It looks like at 6 
months of age, the mutant samples can't really be separated from the WT 
samples either.

- Separation by age is still retained across PC2. However, this was already 
an incredibly strong effect in the gene PCA. 

- This would suggest that the effects on gene expression through the 
Q96_K97del/+ genotype may be too subtle to be able to be captured through 
such a small set of latent variables. 


## 2. Next steps

- Will test out CoGAPs and ProjectR as well just to see if 
non-negative matrix factorisation has any utility at all.

- Seeing as we've previously gotten really clean and interpretable 
results using standard gene set enrichment testing methods with 
Hallmark / KEGG gene set collections, I would say that those approaches 
seem more reliable for our aims. 


**Limitations:**

- I used MultiPLIER with default settings and parameters. The model 
is intended to be easily transferred to any dataset, so would have 
expected it to give some decent results, but it seems it just didn't 
really work here, so not much point further tuning settings. 

- MultiPLIER was created based on human RNA-seq data. Not all human 
genes have zebrafish homologs, so it's possible that the LVs determined 
in humans may not apply to zebrafish. However, considering that 
MSigDB gene sets easily give interpretable results in zebrafish, I'm 
not sure.

- [This](https://www.mdpi.com/2073-4425/11/2/226/htm) study  used 
MultiPLIER to get additional insight into differences between 
tumour subtypes and found it wasn't super informative at distinguishing 
between the subtypes (subtypes had similar distributions of LV expression 
and PCA was similar to original with genes). More analysis at their 
[GitHub](https://github.com/Sage-Bionetworks/NF_LandscapePaper_2019).

