---
title: "PathwayPCA on Mayo Clinic RNA-seq dataset"
author: "Nhi Hin"
date: "2020-06-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r}
library(pathwayPCA)
library(edgeR)
library(here)
library(ggplot2)
library(magrittr)
library(export)
library(reshape2)
library(dplyr)
library(Rtsne)
library(tibble)
library(pheatmap)
library(stats)
library(cluster)
theme_set(theme_bw())

save_pheatmap_pdf <- function(x, filename, width=8, height=11) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

set.seed(123)
```


### 1. Import gene sets

- According to the [PathwayPCA vignette](https://gabrielodom.github.io/pathwayPCA/articles/Supplement2-Importing_Data.html#overview) we need to set up a `pathwayCollection` object of the gene 
sets to be tested. Here I will use the Hallmark collection from MSigDB. 

- The `ens_h_mapped.rds` object was previously created from the Hallmark 
collection .gmt file from MSigDB (v.7.0). The entrezgene IDs were converted 
to Ensembl human IDs using homologous genes from BioMart. 

```{r}
# Hallmark Gene set collection 
hallmarkGenes <- readRDS(here("data", "genesets", "human",
                              "ens_h_mapped.rds"))
hallmarkNames <- names(hallmarkGenes)

hallmarkCol <- CreatePathwayCollection(
  sets_ls = hallmarkGenes,
  TERMS = hallmarkNames
)

# IRE gene sets
ireGenes <- readRDS(here("data", "ireGenes", "human", "ireGenes.rds"))
ireNames <- names(ireGenes)
ireCol <- CreatePathwayCollection(
  sets_ls = ireGenes, TERMS = ireNames
)
```

### 2. Tidy sample/expression matrix

- The expression data must be in tidy format for inputting into 
*PathwayPCA*. 

```{r}
exprs <- readRDS(here("data", "datasets", "mayo", 
                     "dge.rds")) %>%
  cpm(log=TRUE)

samples <- readRDS(here("data", "datasets", "mayo", 
                        "dge.rds"))$samples 
rownames(samples) <- samples$SampleID
samples %<>% rownames_to_column("Sample")

tidyDat <- exprs %>% 
  as.data.frame %>%
  rownames_to_column("gene") %>% 
  TransposeAssay()

tidyDat[1:5, 1:20]  # preview
```


- Subsetting data

```{r}
# Temporal cortex
samples_ADcontrols_t <- samples %>%
  dplyr::filter(Diagnosis %in% c("AD", "Control")) %>%
  dplyr::filter(Tissue == "TemporalCortex")

# Cerebellum
samples_ADcontrols_c <- samples %>%
  dplyr::filter(Diagnosis %in% c("AD", "Control")) %>%
  dplyr::filter(Tissue == "Cerebellum")

tidyDat_t <- tidyDat %>%
  dplyr::filter(Sample %in% samples_ADcontrols_t$SampleID)

tidyDat_c <- tidyDat %>%
  dplyr::filter(Sample %in% samples_ADcontrols_c$SampleID)
```


### 3. `Omics` data object

- The samples/expression tidy data and gene sets must now be placed inside an 
`Omics` object.

- We have to create different `Omics` objects to test different contrasts. 
*PathwayPCA* does not support complex designs like regular gene set testing 
methods utilising *limma* objects. However, they do support the following 
modes:

    - Regression on a continuous variable (e.g. age)
    - Categorical (only binary supported) (e.g. `hasPSEN` mutation)
    - Survival (not relevant to our analysis). 

- Because of this *PathwayPCA* should be able to give us some complementary 
information/results. 

### 4. AES-PCA

- Here we test whether pathways are enriched on the binary variable `Diagnosis` 
(AD or control) using AES-PCA.

- Analyses will be done separately on temporal cortex and cerebellum 
tissues as these tissues are generally very different in their expression 
patterns. 

- **Temporal cortex**

```{r message=FALSE}
# Create the samples dataframe
samples_genotype_t <- samples_ADcontrols_t %>%
  dplyr::select(Sample, Diagnosis) %>%
  dplyr::rename(response = Diagnosis)

# Omics object requires the expression data in tidy form 
# as well as the samples dataframe
omics_genotype_t <- CreateOmics(
  assayData_df = tidyDat_t,
  pathwayCollection_ls = hallmarkCol,
  response = samples_genotype_t,
  respType = "categ"
)
# Create another one for the IRE sets
omics_genotype_t_ire <- CreateOmics(
  assayData_df = tidyDat_t,
  pathwayCollection_ls = ireCol,
  response = samples_genotype_t,
  respType = "categ"
)
omics_genotype_t_ire %>% saveRDS(here("data","datasets", "mayo", 
                                      "omics_genotype_t_ire.rds"))

# Perform enrichment testing using AES-PCA
test_genotype_t <- AESPCA_pVals(
  object = omics_genotype_t,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)
# And for IRE
# AESPCA_pVals(
#   object = omics_genotype_t_ire,
#   numReps = 0,
#   numPCs = 2,
#   parallel = TRUE,
#   numCores = 8,
#   adjustpValues = TRUE,
#   adjustment = c("Holm", "BH")
# )%>% saveRDS("./test_genotype_t_ire.rds")
test_genotype_t_ire <- readRDS(here("data", "datasets", "mayo",
                                    "test_genotype_t_ire.rds"))

# Top ranked enriched gene sets 
test_genotype_t$pVals_df 
test_genotype_t_ire$pVals_df


# Which gene sets are significant?
sigSets_t <- test_genotype_t$pVals_df %>%
  dplyr::filter(FDR_BH < 0.05)

sigSets_t
```


- **Cerebellum**

```{r message=FALSE}
samples_genotype_c <- samples_ADcontrols_c %>%
  dplyr::select(Sample, Diagnosis) %>%
  dplyr::rename(response = Diagnosis)

omics_genotype_c <- CreateOmics(
  assayData_df = tidyDat_c,
  pathwayCollection_ls = hallmarkCol,
  response = samples_genotype_c,
  respType = "categ"
)

test_genotype_c <- AESPCA_pVals(
  object = omics_genotype_c,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

test_genotype_c$pVals_df 

sigSets_c <- test_genotype_c$pVals_df %>%
  dplyr::filter(FDR_BH < 0.05)

sigSets_c
```

### 5. Plots

- Violin plot of the IRE gene sets

```{r}

ire_comparisons <- list(c("Control", "AD"))
violin_ire_plot <- test_genotype_t_ire$PCs_ls %>%
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype_t_ire$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "geneset") %>%
  left_join(samples, by = "Sample") %>%
  ggpubr::ggviolin(x = "Diagnosis", y = "V1",
                    fill = "Diagnosis", add = "boxplot",
                   add.params = list(fill = "white"),
                   palette = c("#dddddd", "orangered")) +
  ggpubr::stat_compare_means(comparisons = ire_comparisons)+
  #ggpubr::stat_compare_means()  +
  facet_wrap(~geneset, scales = "free_y") +
  labs(y = "PC1 Value")

violin_ire_plot %>% export::graph2pdf(here("data", "datasets", 
                                           "mayo", "fig", "violin_ire_plot.pdf"))

# test_genotype_t_ire$PCs_ls$ire3_all %>% 
#   set_rownames(attributes(test_genotype_t_ire$PCs_ls)$sampleIDs) %>%
#   rownames_to_column("Sample") %>%
#   left_join(samples, by = "Sample") %>%
#   ggplot(aes(x = V1, fill = Diagnosis, colour = AgeAtDeath)) + 
#   geom_histogram(bins = 20, position = "identity", alpha=0.5) +
#   labs(x = "PC1 value for each sample",
#        y = "Count", 
#        fill = "Has AD?") +
#   scale_fill_manual(values = c("MediumTurquoise", "orangered")) +
#   scale_colour_manual(values = c("black", "yellow"))+
#   ggtitle("Temporal Cortex: 3' IREs")
# 
# test_genotype_t_ire$PCs_ls$ire5_all %>% 
#   set_rownames(attributes(test_genotype_t_ire$PCs_ls)$sampleIDs) %>%
#   rownames_to_column("Sample") %>%
#   left_join(samples, by = "Sample") %>%
#   ggplot(aes(x = V1, fill = Diagnosis, colour = AgeAtDeath)) + 
#   geom_histogram(bins = 20, position = "identity", alpha=0.5) +
#   labs(x = "PC1 value for each sample",
#        y = "Count", 
#        fill = "Has AD?") +
#   scale_fill_manual(values = c("MediumTurquoise", "orangered")) +
#   scale_colour_manual(values = c("black", "yellow"))+
#   ggtitle("Temporal Cortex: 5' IREs")
# 
# test_genotype_t_ire$PCs_ls$ire3_hq %>% 
#   set_rownames(attributes(test_genotype_t_ire$PCs_ls)$sampleIDs) %>%
#   rownames_to_column("Sample") %>%
#   left_join(samples, by = "Sample") %>%
#   ggplot(aes(x = Diagnosis, y = V1, fill = Diagnosis, colour = AgeAtDeath)) + 
#   geom_boxplot() +
#   #geom_histogram(bins = 20, position = "identity", alpha=0.5) +
#   labs(x = "PC1 value for each sample",
#        y = "Count", 
#        fill = "Has AD?") +
#   scale_fill_manual(values = c("MediumTurquoise", "orangered")) +
#   scale_colour_manual(values = c("black", "yellow"))+
#   ggtitle("Temporal Cortex: High quality 3' IREs")
```

```{r}
test_genotype_c$PCs_ls$HALLMARK_OXIDATIVE_PHOSPHORYLATION %>% 
  set_rownames(attributes(test_genotype_c$PCs_ls)$sampleIDs) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = Diagnosis, colour = AgeAtDeath)) + 
  geom_histogram(bins = 20, position = "identity", alpha=0.5) +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("MediumTurquoise", "orangered")) +
  scale_colour_manual(values = c("black", "yellow"))+
  ggtitle("Cerebellum: Oxidative Phosphorylation")
```


- All 50 of the Hallmark gene sets will be plotted below. 

- **Temporal cortex**:

```{r }
plot_TCenr <- test_genotype_t$PCs_ls %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype_t$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::mutate(pathway = gsub(x = pathway,
                               pattern = "_",
                               replacement = " "),
                pathway = gsub(x = pathway,
                               pattern = "HALLMARK ",
                               replacement = "")) %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = Diagnosis, colour = AgeAtDeath))+
  facet_wrap(~pathway, scales = "free_y", ncol = 5)+
  geom_histogram(bins = 25, position = "identity", alpha=0.5) +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Genotype") +
  scale_fill_manual(values = c("MediumTurquoise", "orangered")) +
  scale_color_manual(values = c("black", "yellow")) 

plot_TCenr

# export::graph2pdf(plot_TCenr, 
#                   here("data", "datasets", "mayo", "fig", "plot_TCenr"), 
#                   width = 15, height=15)
```

- **Cerebellum**:

```{r}
plot_Cerenr <- test_genotype_c$PCs_ls %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype_c$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::mutate(pathway = gsub(x = pathway,
                               pattern = "_",
                               replacement = " "),
                pathway = gsub(x = pathway,
                               pattern = "HALLMARK ",
                               replacement = "")) %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = Diagnosis, colour = AgeAtDeath))+
  facet_wrap(~pathway, scales = "free_y", ncol = 5)+
  geom_histogram(bins = 25, position = "identity", alpha=0.5) +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Genotype") +
  scale_fill_manual(values = c("MediumTurquoise", "orangered")) +
  scale_color_manual(values = c("black", "yellow")) 

plot_Cerenr

# export::graph2pdf(plot_Cerenr, 
#                   here("data", "datasets", "mayo", "fig", "plot_Cerenr"), 
#                   width = 15, height=15)
```


## 6. Patient clustering

- Is it possible to identify clusters of patients based on their PC1 values 
across the pathways?

- First, we will extract the PC1 values for all Temporal Cortex samples 
and their hallmark gene sets. These PC1 values correspond to effects that 
distinguish the Diagnosis (AD from control). 

```{r}
# Format dataframe into wide form suitable for PCA
PC1_TCsamples <- test_genotype_t$PCs_ls %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype_t$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::select(-V2) %>%
  dcast(Sample ~ pathway, value.var = "V1") %>% 
  column_to_rownames("Sample")

head(PC1_TCsamples)

# Perform just normal PCA
PCA_TCsamples <- PC1_TCsamples %>% prcomp()

PCAplot_TCsamples <- PCA_TCsamples$x %>% 
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  ggplot(aes(x=PC1, y = PC2, 
             colour = Diagnosis, 
             shape = ApoE)) +
  geom_point(alpha = 0.6,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (82.5%)",
         y = "Principal Component 2 (10.6%)", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 

PCAplot_TCsamples
```

- From the normal PCA it doesn't look like the AD patients have 
immediately visible clusters. The AD samples seem to group near 
one side of the plot but there is considerable heterogeneity 
amongst the control samples. The ApoE genotype seems to have some 
correlation but this isn't anything exact as certain ApoE alleles 
are over-represented in the data anyway. 

- Let's show the gene set activities as well on a heatmap

```{r fig.height = 11}
# format data
data_for_heatmap <- PC1_TCsamples %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample", "Diagnosis", "AgeAtDeath", "ApoE")])%>%
  dplyr::arrange(Diagnosis, ApoE) %>%
  column_to_rownames("Sample")

annot_for_heatmap <- data_for_heatmap %>%
  dplyr::select(Diagnosis, AgeAtDeath, ApoE)

data_for_heatmap %<>% 
  dplyr::select(-Diagnosis, -AgeAtDeath, -ApoE) %>%
  t%>%
  set_rownames(gsub(x = rownames(.),
                              pattern = "HALLMARK_",
                              replacement = "") )

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)
plotting_heatmap <- pheatmap(data_for_heatmap,
                             cluster_cols = FALSE,
                             scale = "row",
                             border_color = "white",
                             gaps_col = 65,
                             # cellwidth = 10,
                             # cellheight = 10,
                             color = palette,
                             annotation_col = annot_for_heatmap)
plotting_heatmap

# save_pheatmap_pdf(plotting_heatmap,
#                   here("data", "datasets", "mayo", "fig", 
#                        "allSamplesHeatmap.pdf"), width = 11,
#                   height = 11)
```

- The heatmap reflects what we saw in the PCA. Overall, the AD samples do 
seem different from the Controls, but there is considerable 
heterogeneity within the control samples. 

#### 6.2. t-SNE

- Sometimes t-SNE can work better than PCA for visualising groups within a 
dataset, as it is non-linear while PCA works better for linearly separable 
data. 
 
- As t-SNE is influenced by parameter choice, we will try various values. 
Typically, the perplexity ranges from 5 to 50 depending on the size of 
the dataset, with larger values favouring a more global view and 
smaller values favouring more local structure. 

```{r }

tsne_TCsamples <- list()
perplexitiesToTest <- c(5, 10, 20)

for(n in perplexitiesToTest){
  tsne_TCsamples[[n]] <-  Rtsne(PC1_TCsamples, 
                        dims = 2, 
                        perplexity= n, 
                        verbose=TRUE, 
                        theta = 0,
                        max_iter = 1e5)
}

```

- We will plot the different t-SNEs below. 

```{r}
# tsne_TCsamples <- readRDS(here("data", "datasets", "mayo",
#                                 "tsne_TCsamples.rds"))

# 5 distinct clusters, although there are AD and controls within each
tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  ggplot(aes(x=Dim1, y = Dim2, 
             colour = Diagnosis, 
             shape = ApoE)) +
  geom_point(alpha = 0.6,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Dim 1",
         y = "Dim 2", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 

# 4-5 main groupings, some separation of AD and controls 
tsne_TCsamples[[10]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  ggplot(aes(x=Dim1, y = Dim2, 
             colour = Diagnosis, 
             shape = ApoE)) +
  geom_point(alpha = 0.6,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Dim 1",
         y = "Dim 2", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 

# This is probably too high of a value as it is mainly the global 
# structure. One main blob
tsne_TCsamples[[20]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  ggplot(aes(x=Dim1, y = Dim2, 
             colour = Diagnosis, 
             shape = ApoE)) +
  geom_point(alpha = 0.6,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Dim 1",
         y = "Dim 2", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 


```

- Based on the t-SNEs we will try clustering the PC1 values into 4-5 
clusters.

- We will use a clustering method

```{r}
clusters_km <- tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>% 
  kmeans(centers = 5) %>%
  .$cluster %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  set_colnames(c("Sample", "Cluster"))

clusters_hier <- PC1_TCsamples %>%
  dist(method = "euclidean") %>%
  hclust(method = "ward.D2")%>%
  cutree(k = 5) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  set_colnames(c("Sample", "Cluster"))

clusters_hier2 <- PC1_TCsamples %>%
  dist(method = "euclidean") %>%
  hclust(method = "ward.D")%>%
  cutree(k = 6) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  set_colnames(c("Sample", "Cluster"))
```

- Plot the clusters on the t-SNE

```{r}
tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  left_join(clusters_km) %>%
  ggplot(aes(x=Dim1, y = Dim2, 
             colour = Diagnosis,
             shape = ApoE)) +
  geom_point(size = 5, alpha=0.3, colour = clusters_km$Cluster) +
  geom_point(alpha = 0.8,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Dim 1",
         y = "Dim 2", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 

tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  left_join(clusters_hier) %>%
  ggplot(aes(x=Dim1, y = Dim2, 
             colour = Diagnosis,
             shape = ApoE)) +
  geom_point(size = 5, alpha=0.3, colour = clusters_hier$Cluster) +
  geom_point(alpha = 0.8,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Dim 1",
         y = "Dim 2", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 
```

- Try clustering on the t-SNE directly

```{r}
clusters_tsne_km <- tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  as.matrix %>%
  kmeans(centers = 5) %>%
  .$cluster%>%
  as.data.frame() %>% 
  rownames_to_column("Sample") %>%
  set_colnames(c("Sample", "Cluster"))

tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  left_join(clusters_tsne_km) %>%
  ggplot(aes(x=Dim1, y = Dim2, 
             colour = Diagnosis,
             shape = ApoE)) +
  geom_point(size = 5, alpha=0.3, colour = clusters_tsne_km$Cluster) +
  geom_point(alpha = 0.8,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Dim 1",
         y = "Dim 2", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 
```

- Heatmap (works well with just the hierarchical clustering)

```{r}
heatmapplot <- PC1_TCsamples %>% 
  rownames_to_column("Sample") %>% 
  left_join(clusters_km) %>%
  left_join(samples[, c("Sample", "Diagnosis")], by = "Sample")%>%
  dplyr::arrange(Cluster, Diagnosis) %>%
  dplyr::select(-Cluster, -Diagnosis) %>%
  column_to_rownames("Sample") %>% t 

rownames(heatmapplot) <- gsub(x = rownames(heatmapplot),
                              pattern = "HALLMARK_",
                              replacement = "") 

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

heatmapAnnot <- clusters_km%>% as.data.frame%>%
  set_rownames(.$Sample) %>%
  magrittr::extract(colnames(heatmapplot),) %>%
  rownames_to_column("Sample2") %>%
  dplyr::select(-Sample) %>%
  dplyr::rename(Sample = Sample2) %>%
  left_join(samples[, c("Sample", "Diagnosis", "AgeAtDeath", "ApoE")]) %>%
  column_to_rownames("Sample") 

plotHeatmap <- pheatmap::pheatmap(heatmapplot, 
                   cluster_cols = FALSE,
                   scale = "row",
                   color = palette,
                   annotation_col = heatmapAnnot)

heatmapAnnot %>% saveRDS(here("data", "datasets", "mayo","heatmapAnnot.rds"))
```



```{r eval=FALSE}



tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "Sex",
                        "AgeAtDeath",
                        "ApoE",
                        "Diagnosis")], by = "Sample")%>%
  left_join(testclust2) %>%
  ggplot(aes(x=Dim1, y = Dim2, 
             colour = Diagnosis,
             shape = ApoE)) +
  geom_point(size = 5, alpha=0.3, colour = testclust2$Cluster) +
  geom_point(alpha = 0.8,size=3) +
  scale_shape_manual(values = c(15,16,17,18,8)) +
  scale_colour_manual(values = c("#999999", "orangered")) +
  theme(aspect.ratio = 1) +
    labs(x = "Dim 1",
         y = "Dim 2", 
         colour = "Diagnosis", shape = "ApoE\nGenotype") 


testclust <- tsne_TCsamples[[5]]$Y %>%
  as.data.frame %>%
  set_colnames(c("Dim1", "Dim2")) %>%
  set_rownames(rownames(PC1_TCsamples)) %>%
  as.matrix %>%
  kmeans(centers = 5) 

testclust2 <- testclust$cluster%>%
  as.data.frame() %>% 
  rownames_to_column("Sample") %>%
  set_colnames(c("Sample", "Cluster"))



```

## Association between ApoE and Cluster

- Not really an association between ApoE and cluster. 

```{r}
conting <- samples %>%
  dplyr::filter(Tissue == "TemporalCortex", Diagnosis %in% c("AD", "Control")) %>%
  left_join(clusters_km, by = "Sample")%>%
  dplyr::group_by(Cluster, ApoE) %>%
  dplyr::summarise(n = n()) %>%
  ungroup() %>%
  dplyr::mutate(Cluster = as.factor(as.character(Cluster)),
                ApoE = as.factor(as.character(ApoE))) %>% 
  dcast(Cluster ~ ApoE, value.var = "n")

# conting <- samples_mayo %>%
#   left_join(mayo_samples[, c("SampleID", "AgeAtDeath", "ApoE")], 
#             by = c("Sample"="SampleID")) %>% 
#   dplyr::group_by(Cluster,ApoE) %>%
#   dplyr::summarise(n = n())%>%
#   ungroup() %>%
#   dplyr::mutate(Cluster = as.factor(as.character(Cluster)),
#                 ApoE = as.factor(as.character(ApoE))) %>% 
#   dcast(Cluster ~ ApoE, value.var = "n")

conting[is.na(conting)] <- 0

# conting2 <- conting %>% 
#   mutate(Allele2 = `23` + `24`,
#          Allele3 = `23` + `33` + `34`,
#          Allele4 = `34` + `44`) %>%
#   dplyr::select(Cluster, Allele2, Allele3, Allele4)

conting %<>% column_to_rownames("Cluster")
# conting2 %<>% column_to_rownames("Cluster")

fisher.test(conting, simulate.p.value = TRUE, B = 1e5) 

```

Different ages between clusters?

```{r}
samples %>% 
  dplyr::filter(Diagnosis %in% c("AD", "Control"), 
                Tissue == "TemporalCortex")%>%
  left_join(clusters_km, by = "Sample")%>%
  dplyr::group_by(Cluster, AgeAtDeath, Diagnosis) %>%
  dplyr::summarise(n = n())%>%
  ungroup() %>%
  dplyr::mutate(Cluster = as.factor(as.character(Cluster))) %>%
  ggplot(aes(x = Cluster, y = AgeAtDeath, colour = Diagnosis)) +
  geom_boxplot() 

samples %>%
  left_join(mayo_samples[, c("SampleID", "AgeAtDeath", "ApoE", "Diagnosis")], 
            by = c("Sample"="SampleID")) %>%
  dplyr::group_by(Cluster, AgeAtDeath, Diagnosis) %>%
  dplyr::summarise(n = n())%>%
  ungroup() %>%
  dplyr::mutate(Cluster = as.factor(as.character(Cluster))) %>%
  ggplot(aes(x = Cluster, y = AgeAtDeath, colour = Diagnosis)) +
  geom_boxplot() 
```



## Export Objects

- PC1 values

```{r eval=FALSE}
PC1_TCsamples %>% saveRDS(here("data", "datasets", "mayo", 
                               "PC1_TCsamples.rds"))
```

- Hierarchical clusters

```{r}
clusters_hier %>% saveRDS(here("data", "datasets", "mayo", 
                               "clusters_hier.rds"))
```

```{r}
clusters_km %>% saveRDS(here("data", "datasets", "mayo", "clusters_km.rds"))
```


- PCA of PC1 values

```{r eval=FALSE}
PCAplot_TCsamples%>% export::graph2pdf(here("data", "datasets", 
                                            "mayo", "fig", "PC1_pca.pdf"))
```

- t-SNE object (for reproducing plots

```{r}
tsne_TCsamples %>% saveRDS(here("data", "datasets", "mayo",
                                "tsne_TCsamples.rds"))
```

- Heatmap

```{r eval=FALSE}
save_pheatmap_pdf(plotHeatmap, width = 11, height = 11,
                  filename = here("data", "datasets", "mayo",
                                  "fig", "clusterHeatmap_hier.pdf"))
```

