---
title: "PathwayPCA on fAD-like zebrafish"
author: "Nhi Hin"
date: "2020-06-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r}
library(pathwayPCA)
library(edgeR)
library(here)
library(ggplot2)
library(magrittr)
library(dplyr)
library(tibble)

save_pheatmap_pdf <- function(x, filename, width=8, height=11) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}
```


### 1. Import gene sets

- According to the [PathwayPCA vignette](https://gabrielodom.github.io/pathwayPCA/articles/Supplement2-Importing_Data.html#overview) we need to set up a `pathwayCollection` object of the gene 
sets to be tested. Here I will use the Hallmark collection from MSigDB. 

- The `ens_h_mapped.rds` object was previously created from the Hallmark 
collection .gmt file from MSigDB (v.7.0). The entrezgene IDs were converted 
to Ensembl mouse IDs using homologous genes from BioMart. 

```{r}
hallmarkGenes <- readRDS(here("data", "genesets", "zebrafish",
                              "ens_h_mapped.rds"))
hallmarkNames <- names(hallmarkGenes)

hallmarkCol <- CreatePathwayCollection(
  sets_ls = hallmarkGenes,
  TERMS = hallmarkNames
)
```

### 2. Tidy sample/expression matrix

- The expression data must be in tidy format for inputting into 
*PathwayPCA*. 

```{r}
exprs <- readRDS(here("data", "datasets", "zebrafish", 
                      "dge.rds")) %>%
  cpm(log=TRUE)

samples <- readRDS(here("data", "datasets", "zebrafish", 
                        "dge.rds"))$samples
samples %<>% rownames_to_column("Sample")

tidyDat <- exprs %>% 
  as.data.frame %>%
  rownames_to_column("gene") %>% 
  TransposeAssay()

tidyDat[1:5, 1:20]  # preview
```

### 3. `Omics` data object

- The samples/expression tidy data and gene sets must now be placed inside an 
`Omics` object.

- We have to create different `Omics` objects to test different contrasts. 
*PathwayPCA* does not support complex designs like regular gene set testing 
methods utilising *limma* objects. However, they do support the following 
modes:

    - Regression on a continuous variable (e.g. age)
    - Categorical (only binary supported) (e.g. `hasPSEN` mutation)
    - Survival (not relevant to our analysis). 

- Because of this *PathwayPCA* should be able to give us some complementary 
information/results. 

### 4. AES-PCA

**Has AD**

- Here we test whether pathways are enriched on the binary variable `genotype` 
using AES-PCA

```{r}
samples_genotype <- samples %>%
  dplyr::select(Sample, Genotype) %>%
  dplyr::rename(response = Genotype)

omics_genotype <- CreateOmics(
  assayData_df = tidyDat,
  pathwayCollection_ls = hallmarkCol,
  response = samples_genotype,
  respType = "categ"
)

# Perform enrichment testing using AES-PCA
test_genotype <- AESPCA_pVals(
  object = omics_genotype,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

test_genotype$pVals_df 

sigSets <- test_genotype$pVals_df %>%
  dplyr::filter(FDR_BH < 0.05)

sigSets

# Perform enrichment testing using Supervised PCA
# test_AD2 <- SuperPCA_pVals(
#   object = omics_AD,
#   numPCs = 2,
#   parallel = TRUE,
#   numCores = 2,
#   adjustpValues = TRUE,
#   adjustment = c("SidakSS", "BY")
# )
# test_AD2$pVals_df
```

### 5. Plots

 - **Reactive oxygen species pathway**: Samples with AD tend to have lower 
 expression of genes in this pathway compared to samples without AD. 

```{r}
test_genotype$PCs_ls$HALLMARK_OXIDATIVE_PHOSPHORYLATION %>% 
  set_rownames(attributes(test_genotype$PCs_ls)$sampleIDs) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  dplyr::filter(Hypoxia == 0) %>%
  ggplot(aes(x = V1, fill = Genotype, colour = Age)) + 
  geom_histogram(bins = 8) +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("#bbbbbb", "orangered")) +
  scale_colour_manual(values = c("black", "yellow"))+
  ggtitle("Oxidative Phosphorylation")
```

#### 5.1. Boxplots

- All Hallmark gene sets as boxplots

```{r}
#scaleFUN <- function(x) sprintf("%.2f", x)
roundSigFig <- function(x){
  signif(x, 1)
}

PC1_plot <- test_genotype$PCs_ls %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::mutate(pathway = gsub(x = pathway,
                               pattern = "_",
                               replacement = " "),
                pathway = gsub(x = pathway,
                               pattern = "HALLMARK ",
                               replacement = "")) %>%
  left_join(samples, by = "Sample") %>%
  dplyr::filter(Hypoxia == 0) %>%
  ggplot(aes(x = Group, y = V1, fill = Group))+
  facet_wrap(~pathway, scales = "free_y", ncol = 5)+
  geom_boxplot() +
  #geom_histogram(bins = 10, position = "identity", alpha=0.6) +
  labs(x = "Group",
       y = "PC1 Value", 
       fill = "Group") +
  scale_fill_manual(values = c( "orangered", "LightCoral",
                                "RoyalBlue", "SkyBlue")) +
  scale_y_continuous(labels=roundSigFig)+
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

PC1_plot
```

#### 5.2. PCA


- Get the PC1 values into wide format 

```{r}
PC1_allSamples <- test_genotype$PCs_ls %>%
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::select(-V2) %>%
  dcast(Sample ~ pathway, value.var = "V1") %>% 
  column_to_rownames("Sample")

head(PC1_allSamples)
```

- Perform PCA to see if there's any structure / grouping in the data

```{r}
PCA_allSamples <- PC1_allSamples %>% prcomp()

PCAplot_allSamples <- PCA_allSamples$x %>%
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by ="Sample") %>%
  ggplot(aes(x = PC1, y = PC2,
             colour = Group,
             shape = paste0(Age, "_", Hypoxia))) + 
  geom_point(alpha = 0.6, size = 3) +
  scale_shape_manual(values = c(15,0,16,1))+
  scale_colour_manual(values = c(rep("orangered",4),
                                 rep("#999999", 4))) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (85.3%)",
         y = "Principal Component 2 (9.8%)", 
         colour = "Group", shape = "Age and\nHypoxia") 

PCAplot_allSamples
```


```{r}
# format data
data_for_heatmap <- PC1_allSamples %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample", "Group", "Age", "Genotype", "Hypoxia",
                        "Gender")])%>%
  dplyr::arrange(Age, Hypoxia, Genotype) %>%
  dplyr::filter(Age==6)

annot_for_heatmap <- data_for_heatmap %>%
  as.data.frame %>%
  dplyr::select(Sample, Group, Age, Genotype, Hypoxia, Gender) %>%
  dplyr::arrange(Age, Hypoxia, Genotype)%>%
  column_to_rownames("Sample")
  

data_for_heatmap %<>% 
    column_to_rownames("Sample") %>%
  dplyr::select(-Group, -Age, -Genotype, -Hypoxia, -Gender) %>%
  t%>%
  set_rownames(gsub(x = rownames(.),
                              pattern = "HALLMARK_",
                              replacement = "") )

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

plot_heatmap <- pheatmap::pheatmap(data_for_heatmap,
                   cluster_cols = FALSE,
                   scale="row",
                   #gaps_col=seq(0,32,by=8),
                   gaps_col=8,
                   color = palette,
                   annotation_col = annot_for_heatmap,
                   border_color = "white", 
                   cellwidth = 10, cellheight=10
                   )

```


```{r}
# format data
data_for_heatmap <- PC1_allSamples %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample", "Group", "Age", "Genotype", "Hypoxia",
                        "Gender")])%>%
  dplyr::arrange(Age, Hypoxia, Genotype) %>%
  dplyr::filter(Age==24)

annot_for_heatmap <- data_for_heatmap %>%
  as.data.frame %>%
  dplyr::select(Sample, Group, Age, Genotype, Hypoxia, Gender) %>%
  dplyr::arrange(Age, Hypoxia, Genotype)%>%
  column_to_rownames("Sample")
  

data_for_heatmap %<>% 
    column_to_rownames("Sample") %>%
  dplyr::select(-Group, -Age, -Genotype, -Hypoxia, -Gender) %>%
  t%>%
  set_rownames(gsub(x = rownames(.),
                              pattern = "HALLMARK_",
                              replacement = "") )

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

plot_heatmap <- pheatmap::pheatmap(data_for_heatmap,
                   cluster_cols = FALSE,
                   scale="row",
                   #gaps_col=seq(0,32,by=8),
                   gaps_col=8,
                   color = palette,
                   annotation_col = annot_for_heatmap,
                   border_color = "white", 
                   cellwidth = 10, cellheight=10
                   )


```


## Export objects

- Boxplots of PC1 values 

```{r}
export::graph2pdf(PC1_plot, here("data", "datasets", "zebrafish", 
                                 "fig", "PC1_plot.pdf"), 
                  width = 11, height=20)
```

- PCA of PC1 values 

```{r}
PCAplot_allSamples %>% export::graph2pdf(here("data", "datasets",
                                              "zebrafish", "fig",
                                              "PC1_pca.pdf"))
```

- Heatmaps of PC1 values for 6 and 24 months

```{r}
save_pheatmap_pdf(plot_heatmap, here("data", "datasets", "zebrafish", "fig", 
                                     "heatmap_6mth.pdf"), 
                  width = 11, height = 11)

save_pheatmap_pdf(plot_heatmap, here("data", "datasets", "zebrafish", "fig", 
                                     "heatmap_24mth.pdf"), 
                  width = 11, height = 11)
```

