---
title: "PathwayPCA on 5XFAD Mouse Dataset"
author: "Nhi Hin"
date: "2020-06-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r}
library(pathwayPCA)
library(edgeR)
library(here)
library(ggplot2)
library(magrittr)
library(dplyr)
library(tibble)

save_pheatmap_pdf <- function(x, filename, width=8, height=11) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}
```


### 1. Import gene sets

- According to the [PathwayPCA vignette](https://gabrielodom.github.io/pathwayPCA/articles/Supplement2-Importing_Data.html#overview) we need to set up a `pathwayCollection` object of the gene 
sets to be tested. Here I will use the Hallmark collection from MSigDB. 

- The `ens_h_mapped.rds` object was previously created from the Hallmark 
collection .gmt file from MSigDB (v.7.0). The entrezgene IDs were converted 
to Ensembl mouse IDs using homologous genes from BioMart. 

```{r}
hallmarkGenes <- readRDS(here("data", "genesets", "mouse",
                              "ens_h_mapped.rds"))
hallmarkNames <- names(hallmarkGenes)

hallmarkCol <- CreatePathwayCollection(
  sets_ls = hallmarkGenes,
  TERMS = hallmarkNames
)

ireGenes <- readRDS(here("data", "ireGenes", "mouse", "ireGenes.rds"))
ireNames <- names(ireGenes)
ireCol <- CreatePathwayCollection(
  sets_ls = ireGenes, TERMS = ireNames
)
```

### 2. Tidy sample/expression matrix

- The expression data must be in tidy format for inputting into 
*PathwayPCA*. 

- We will perform *PathwayPCA* separately for each age group in the 
dataset. 

```{r}
exprs <- readRDS(here("data", "datasets", "mouse", 
                      "GSE140286", "dge.rds")) %>%
  cpm(log=TRUE)

samples <- readRDS(here("data", "datasets", "mouse", 
                        "GSE140286", "dge.rds"))$samples %>%
  rownames_to_column("Sample") 
samples$genotype %<>% relevel( "WT") 

# Subset different age groups
# 3 month olds
samples_3mth <- samples %>% 
  dplyr::filter(age == 3)
exprs_3mth <- exprs[, samples_3mth$Sample]

# 6 month olds
samples_6mth <- samples %>% 
  dplyr::filter(age == 6)
exprs_6mth <- exprs[, samples_6mth$Sample]

# 12 month olds
samples_12mth <- samples %>% 
  dplyr::filter(age == 12)
exprs_12mth <- exprs[, samples_12mth$Sample]

# tidyData objects for each age group
tidyDat_3mth <- exprs_3mth %>%
  as.data.frame %>%
  rownames_to_column("gene") %>%
  TransposeAssay()

tidyDat_6mth <- exprs_6mth %>%
  as.data.frame %>%
  rownames_to_column("gene") %>%
  TransposeAssay()

tidyDat_12mth <- exprs_12mth %>%
  as.data.frame %>%
  rownames_to_column("gene") %>%
  TransposeAssay()
```

### 3. `Omics` data object

- The samples/expression tidy data and gene sets must now be placed inside an 
`Omics` object.

- We have to create different `Omics` objects to test different contrasts. 
*PathwayPCA* does not support complex designs like regular gene set testing 
methods utilising *limma* objects. However, they do support the following 
modes:

    - Regression on a continuous variable (e.g. age)
    - Categorical (only binary supported) (e.g. `hasPSEN` mutation)
    - Survival (not relevant to our analysis). 

- Because of this *PathwayPCA* should be able to give us some complementary 
information/results. 

### 4. AES-PCA

**Has AD**

- Here we test whether pathways are enriched on the binary variable `genotype` 
using AES-PCA

```{r}
# The CreateOmics function will only take a response dataframe with one 
# column so we need to subset the sample tables for the different ages.
samples_genotype_3mth <- samples_3mth %>%
  dplyr::select(Sample, genotype) %>%
  dplyr::rename(response = genotype)

samples_genotype_6mth <- samples_6mth %>%
  dplyr::select(Sample, genotype) %>%
  dplyr::rename(response = genotype)

samples_genotype_12mth <- samples_12mth %>%
  dplyr::select(Sample, genotype) %>%
  dplyr::rename(response = genotype)

# Create omics object for each age group
omics_genotype_3mth <- CreateOmics(
  assayData_df = tidyDat_3mth,
  pathwayCollection_ls = hallmarkCol,
  response = samples_genotype_3mth,
  respType = "categ"
)

omics_genotype_6mth <- CreateOmics(
  assayData_df = tidyDat_6mth,
  pathwayCollection_ls = hallmarkCol,
  response = samples_genotype_6mth,
  respType = "categ"
)

omics_genotype_12mth <- CreateOmics(
  assayData_df = tidyDat_12mth,
  pathwayCollection_ls = hallmarkCol,
  response = samples_genotype_12mth,
  respType = "categ"
)

# Perform enrichment testing using AES-PCA
test_genotype_3mth <- AESPCA_pVals(
  object = omics_genotype_3mth,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

test_genotype_6mth <- AESPCA_pVals(
  object = omics_genotype_6mth,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

test_genotype_12mth <- AESPCA_pVals(
  object = omics_genotype_12mth,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

# Look at p-values
test_genotype_3mth$pVals_df 
test_genotype_6mth$pVals_df 
test_genotype_12mth$pVals_df 

# Extract significant sets
sigSets_3mth <- test_genotype_3mth$pVals_df %>%
  dplyr::filter(FDR_BH < 0.05)
sigSets_6mth <- test_genotype_6mth$pVals_df %>%
  dplyr::filter(FDR_BH < 0.05)
sigSets_12mth <- test_genotype_12mth$pVals_df %>%
  dplyr::filter(FDR_BH < 0.05)
```

#### 4.1. Comparison to conventional GSEA results

```{r}
mouResults <- readRDS(here("data", "datasets", "mouse", 
                            "GSE140286", "gsea.rds"))
```

- Conventional GSEA found that 16 gene sets were enriched. 

```{r}
mouResults$combTest$effectOf5XFAD_3mth %>% dplyr::filter(fdr < 0.05) 
```

- In contrast the AES-PCA method found that no gene sets were enriched. 
This is likely because of the small sample size (and performing AES-PCA on 
this age group separately - as opposed to borrowing info in the GSEA 
approach which utilises *limma*). 

```{r}
sigSets_3mth
```

- However, if we look at the 6 month age group (49 gene sets enriched)

```{r}
mouResults$combTest$effectOf5XFAD_6mth %>% dplyr::filter(fdr< 0.05) 
```

- 44 gene sets enriched

```{r}
sigSets_6mth
```

```{r}
sigSets_6mth %>% inner_join(mouResults$combTest$effectOf5XFAD_6mth %>% 
                              dplyr::filter(fdr< 0.05), 
                            by = c("pathways"="Geneset") ) 
```


```{r}
library(eulerr)
mouseVenn <- c("GSEA"=49, "AESPCA"=44, "GSEA&AESPCA"=44)
eulerr::venn(mouseVenn) %>% plot(quantities=TRUE) %>%
  export::graph2pdf(here("data", "datasets", "mouse", "GSE140286", "fig", 
                         "venn.pdf"))
```


- IRE

```{r}
# Create omics object for each age group
omics_genotype_3mth_ire <- CreateOmics(
  assayData_df = tidyDat_3mth,
  pathwayCollection_ls = ireCol,
  response = samples_genotype_3mth,
  respType = "categ"
)
omics_genotype_3mth_ire %>% saveRDS(here("data", "datasets", "mouse", 
                                         "GSE140286", 
                                         "omics_genotype_3mth_ire.rds"))

omics_genotype_6mth_ire <- CreateOmics(
  assayData_df = tidyDat_6mth,
  pathwayCollection_ls = ireCol,
  response = samples_genotype_6mth,
  respType = "categ"
)
omics_genotype_6mth_ire %>% saveRDS(here("data", "datasets", "mouse", 
                                         "GSE140286", 
                                         "omics_genotype_6mth_ire.rds"))

omics_genotype_12mth_ire <- CreateOmics(
  assayData_df = tidyDat_12mth,
  pathwayCollection_ls = ireCol,
  response = samples_genotype_12mth,
  respType = "categ"
)
omics_genotype_12mth_ire %>% saveRDS(here("data", "datasets", "mouse", 
                                         "GSE140286", 
                                         "omics_genotype_12mth_ire.rds"))


AESPCA_pVals(
  object = omics_genotype_3mth_ire,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 8,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
) %>% saveRDS("./test_genotype_3mth_ire.rds")
test_genotype_3mth_ire <- readRDS(here("data", "datasets", "mouse",
                                       "GSE140286", 
                                       "test_genotype_3mth_ire.rds"))

AESPCA_pVals(
  object = omics_genotype_6mth_ire,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 8,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
) %>% saveRDS("./test_genotype_6mth_ire.rds")
test_genotype_6mth_ire <- readRDS(here("data", "datasets", "mouse",
                                       "GSE140286", 
                                       "test_genotype_6mth_ire.rds"))

AESPCA_pVals(
  object = omics_genotype_12mth_ire,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 8,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
) %>% saveRDS("./test_genotype_12mth_ire.rds")
test_genotype_12mth_ire <- readRDS(here("data", "datasets", "mouse",
                                       "GSE140286", 
                                       "test_genotype_12mth_ire.rds"))
```

### 5. Gene Loadings

- We can look at the gene loadings of the calculated PCs in order to see 
which genes are contributing most to the activity of a particular gene set. 

- First retrieve gene annotations from `AnnotationHub`.

```{r}
# Search for and load relevant AnnotationHub object:
ah <- AnnotationHub()
# ah %>%
#   subset(grepl("musculus", species)) %>%
#   subset(rdataclass == "EnsDb")
ensDb <- ah[["AH69210"]] # Ensembl 96 EnsDb for Mus musculus

# Get gene information:
genes <- genes(ensDb)
genes %<>% as.data.frame
head(genes)
```

- Test out getting the gene loadings from one gene set (Oxidative 
Phosphorylation). There are 10 genes which are contributing to the 
loading of PC1. 

```{r}
test_genotype_3mth$loadings_ls$HALLMARK_OXIDATIVE_PHOSPHORYLATION %>%
  as.data.frame %>%
  rownames_to_column("gene_id") %>%
  dplyr::filter(PC1 != 0) %>% 
  left_join(genes[, c("gene_id", "gene_name", 
                      "gene_biotype", "description")], by = "gene_id") %>% 
  dplyr::arrange(desc(abs(PC1))) 
```

- Apply the above to all gene sets.

```{r}
geneLoadings_3mth <- test_genotype_3mth$loadings_ls %>%
  lapply(function(x){
    x %>%
      as.data.frame %>%
      rownames_to_column("gene_id") %>%
      dplyr::filter(PC1 != 0) %>% 
      left_join(genes[, c("gene_id", "gene_name", 
                      "gene_biotype", "description")], by = "gene_id") %>% 
      dplyr::arrange(desc(abs(PC1))) 
  })

geneLoadings_6mth <- test_genotype_6mth$loadings_ls %>%
  lapply(function(x){
    x %>%
      as.data.frame %>%
      rownames_to_column("gene_id") %>%
      dplyr::filter(PC1 != 0) %>% 
      left_join(genes[, c("gene_id", "gene_name", 
                      "gene_biotype", "description")], by = "gene_id") %>% 
      dplyr::arrange(desc(abs(PC1))) 
  })

geneLoadings_12mth <- test_genotype_12mth$loadings_ls %>%
  lapply(function(x){
    x %>%
      as.data.frame %>%
      rownames_to_column("gene_id") %>%
      dplyr::filter(PC1 != 0) %>% 
      left_join(genes[, c("gene_id", "gene_name", 
                      "gene_biotype", "description")], by = "gene_id") %>% 
      dplyr::arrange(desc(abs(PC1))) 
  })

geneLoadings <- list(
  XFADVsWT_3mth = geneLoadings_3mth,
  XFADvsWT_6mth = geneLoadings_6mth,
  XFADvsWT_12mth = geneLoadings_12mth
)
```

- Loadings for IRE gene sets

```{r}
geneLoadings_ire_3mth <- test_genotype_3mth_ire$loadings_ls%>%
  lapply(function(x){
    x %>%
      as.data.frame %>%
      rownames_to_column("gene_id") %>%
      dplyr::filter(PC1 != 0) %>% 
      left_join(genes[, c("gene_id", "gene_name", 
                      "gene_biotype", "description")], by = "gene_id") %>% 
      dplyr::arrange(desc(abs(PC1))) 
  })

geneLoadings_ire_6mth <- test_genotype_6mth_ire$loadings_ls%>%
  lapply(function(x){
    x %>%
      as.data.frame %>%
      rownames_to_column("gene_id") %>%
      dplyr::filter(PC1 != 0) %>% 
      left_join(genes[, c("gene_id", "gene_name", 
                      "gene_biotype", "description")], by = "gene_id") %>% 
      dplyr::arrange(desc(abs(PC1))) 
  })

geneLoadings_ire_12mth <- test_genotype_12mth_ire$loadings_ls%>%
  lapply(function(x){
    x %>%
      as.data.frame %>%
      rownames_to_column("gene_id") %>%
      dplyr::filter(PC1 != 0) %>% 
      left_join(genes[, c("gene_id", "gene_name", 
                      "gene_biotype", "description")], by = "gene_id") %>% 
      dplyr::arrange(desc(abs(PC1))) 
  })

geneLoadings_ire <- list(
  XFADVsWT_3mth = geneLoadings_ire_3mth,
  XFADvsWT_6mth = geneLoadings_ire_6mth,
  XFADvsWT_12mth = geneLoadings_ire_12mth
)

```

```{r}
geneLoadings %>% saveRDS(here("data", "datasets", "mouse", "GSE140286", 
                              "geneLoadings.rds"))

geneLoadings_ire %>% saveRDS(here("data", "datasets", "mouse", "GSE140286", 
                                  "geneLoadings_ire.rds"))
```


### 5. Visualisations

#### Get PC1 values into wide format

- In order to make plots/visualisations it would be easier to get the PC1 
values into the same format as a typical counts matrix. 

```{r}
# PC1 values for 3 month olds
PC1_3mths <- test_genotype_3mth$PCs_ls %>%
   lapply(function(x){
    x %>% set_rownames(attributes(test_genotype_3mth$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::select(-V2) %>%
  dcast(Sample ~ pathway, value.var = "V1") %>% 
  column_to_rownames("Sample")
head(PC1_3mths)

# PC1 values for 6 month olds
PC1_6mths <- test_genotype_6mth$PCs_ls %>%
   lapply(function(x){
    x %>% set_rownames(attributes(test_genotype_6mth$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::select(-V2) %>%
  dcast(Sample ~ pathway, value.var = "V1") %>% 
  column_to_rownames("Sample")
head(PC1_6mths)

# PC1 values for 12 month olds
PC1_12mths <- test_genotype_12mth$PCs_ls %>%
   lapply(function(x){
    x %>% set_rownames(attributes(test_genotype_12mth$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::select(-V2) %>%
  dcast(Sample ~ pathway, value.var = "V1") %>% 
  column_to_rownames("Sample")
head(PC1_24mths)
```


#### PCA

- Perform PCA to see if there's any structure / grouping in the data

- 3 months (unclear separation between mutant and wt)

```{r}
PCA_3mths <- PC1_3mths %>% prcomp()
summary(PCA_3mths) #PC1 87.3% PC2 10.6%

PCAplot_3mths <- PCA_3mths$x %>%
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by ="Sample") %>%
  ggplot(aes(x = PC1, y = PC2,
             colour = paste0(age, "_", genotype))) + 
  geom_point(alpha = 0.6, size = 3) +
  scale_colour_manual(values = c( "orangered", "#555555")) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (87.3%)",
         y = "Principal Component 2 (10.6%)", 
         colour = "Group", shape = "Group") 
PCAplot_3mths
```

- 6 months (hard to tell because only a few samples, but possibly some 
separation across PC1)

```{r}
PCA_6mths <- PC1_6mths %>% prcomp()
summary(PCA_6mths) #PC1 87.3% PC2 6.4%

PCAplot_6mths <- PCA_6mths$x %>%
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by ="Sample") %>%
  ggplot(aes(x = PC1, y = PC2,
             colour = paste0(age, "_", genotype))) + 
  geom_point(alpha = 0.6, size = 3) +
  scale_colour_manual(values = c( "orangered", "#555555")) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (87.3%)",
         y = "Principal Component 2 (6.4%)", 
         colour = "Group", shape = "Group") 
PCAplot_6mths
```

- 12 months (separation across PC1)

```{r}
PCA_12mths <- PC1_12mths %>% prcomp()
summary(PCA_12mths) #PC1 95.9% PC2 2.0%

PCAplot_12mths <- PCA_12mths$x %>%
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by ="Sample") %>%
  ggplot(aes(x = PC1, y = PC2,
             colour = paste0(age, "_", genotype))) + 
  geom_point(alpha = 0.6, size = 3) +
  scale_colour_manual(values = c( "orangered", "#555555")) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (95.6%)",
         y = "Principal Component 2 (2.0%)", 
         colour = "Group", shape = "Group") 
PCAplot_12mths
```

#### Heatmaps

- 3 months

```{r}
# format data
data_for_heatmap_3mths <- PC1_3mths %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample", "genotype", "sex")])%>%
  dplyr::arrange(genotype) 

annot_for_heatmap_3mths <- data_for_heatmap_3mths %>%
  as.data.frame %>%
  dplyr::select(Sample, genotype, sex) %>%
  dplyr::arrange(genotype)%>%
  column_to_rownames("Sample")
  
data_for_heatmap_3mths %<>% 
    column_to_rownames("Sample") %>%
  dplyr::select(-genotype, -sex) %>%
  t%>%
  set_rownames(gsub(x = rownames(.),
                              pattern = "HALLMARK_",
                              replacement = "") )

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

plot_heatmap_3mth <- pheatmap::pheatmap(data_for_heatmap_3mths,
                   cluster_cols = FALSE,
                   #scale="row",
                   #gaps_col=seq(0,32,by=8),
                   gaps_col=2,
                   color = palette,
                   annotation_col = annot_for_heatmap_3mths,
                   border_color = "white", 
                   cellwidth = 10, cellheight=10
                   )
```

- 6 months

```{r}
# format data
data_for_heatmap_6mths <- PC1_6mths %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample", "genotype", "sex")])%>%
  dplyr::arrange(genotype) 

annot_for_heatmap_6mths <- data_for_heatmap_6mths %>%
  as.data.frame %>%
  dplyr::select(Sample, genotype, sex) %>%
  dplyr::arrange(genotype)%>%
  column_to_rownames("Sample")
  
data_for_heatmap_6mths %<>% 
    column_to_rownames("Sample") %>%
  dplyr::select(-genotype, -sex) %>%
  t%>%
  set_rownames(gsub(x = rownames(.),
                              pattern = "HALLMARK_",
                              replacement = "") )

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

plot_heatmap_6mth <- pheatmap::pheatmap(data_for_heatmap_6mths,
                   cluster_cols = FALSE,
                   #scale="row",
                   #gaps_col=seq(0,32,by=8),
                   gaps_col=3,
                   color = palette,
                   annotation_col = annot_for_heatmap_6mths,
                   border_color = "white", 
                   cellwidth = 10, cellheight=10
                   )
```

- 12 months

```{r}
# format data
data_for_heatmap_12mths <- PC1_12mths %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample", "genotype", "sex")])%>%
  dplyr::arrange(genotype) 

annot_for_heatmap_12mths <- data_for_heatmap_12mths %>%
  as.data.frame %>%
  dplyr::select(Sample, genotype, sex) %>%
  dplyr::arrange(genotype)%>%
  column_to_rownames("Sample")
  
data_for_heatmap_12mths %<>% 
    column_to_rownames("Sample") %>%
  dplyr::select(-genotype, -sex) %>%
  t%>%
  set_rownames(gsub(x = rownames(.),
                              pattern = "HALLMARK_",
                              replacement = "") )

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

plot_heatmap_12mth <- pheatmap::pheatmap(data_for_heatmap_12mths,
                   cluster_cols = FALSE,
                   #scale="row",
                   #gaps_col=seq(0,32,by=8),
                   gaps_col=2,
                   color = palette,
                   annotation_col = annot_for_heatmap_12mths,
                   border_color = "white", 
                   cellwidth = 10, cellheight=10
                   )
```


#### Initial Histogram

```{r}
test_genotype$PCs_ls$HALLMARK_INTERFERON_ALPHA_RESPONSE %>% 
  set_rownames(attributes(test_genotype$PCs_ls)$sampleIDs) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = genotype)) + 
  geom_histogram(bins = 8, colour = "white") +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("#bbbbbb", "orangered")) +
  ggtitle("Interferon Alpha Response")
```

#### Histograms

- All significant gene sets (histograms)

```{r}
test_genotype$PCs_ls[sigSets$pathways] %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::mutate(pathway = gsub(x = pathway,
                               pattern = "_",
                               replacement = " "),
                pathway = gsub(x = pathway,
                               pattern = "HALLMARK ",
                               replacement = "")) %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = genotype)) + 
  geom_histogram(bins = 8, colour = "white") +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Genotype") +
  scale_fill_manual(values = c( "orangered", "#bbbbbb")) +
  facet_wrap(~pathway, scales = "free_y")
  
```

#### Boxplots

- As boxplots 

```{r}
roundSigFig <- function(x){
  signif(x, 1)
}

PC1_plot <- test_genotype$PCs_ls %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_genotype$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::mutate(pathway = gsub(x = pathway,
                               pattern = "_",
                               replacement = " "),
                pathway = gsub(x = pathway,
                               pattern = "HALLMARK ",
                               replacement = "")) %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = paste0(age,"_",genotype), y = V1,
             fill = paste0(age,"_",genotype)))+
  facet_wrap(~pathway, scales = "free_y", ncol = 5)+
  geom_boxplot() +
  #geom_histogram(bins = 10, position = "identity", alpha=0.6) +
  labs(x = "Genotype",
       y = "PC1 Value", 
       fill = "Genotype") +
  # 3 age groups and 2 genotypes so need 6 colours
  scale_fill_manual(values = c( "orangered", "LightCoral", 
                                "RoyalBlue", "SkyBlue",
                                "ForestGreen", "LightGreen")) +
  scale_y_continuous(labels=roundSigFig)+
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

PC1_plot
```


## Export Objects

- PC1 values

```{r}
# PC1_allSamples %>% saveRDS(here("data", "datasets", 
#                                               "mouse", "GSE140286", 
#                                                "PC1.rds"))

PC1_3mths %>% saveRDS(here("data", "datasets", "mouse", "GSE140286",
                           "PC1_3mths.rds"))
PC1_6mths %>% saveRDS(here("data", "datasets", "mouse", "GSE140286",
                           "PC1_6mths.rds"))
PC1_12mths %>% saveRDS(here("data", "datasets", "mouse", "GSE140286",
                           "PC1_12mths.rds"))
data_for_heatmap_12mths  %>% saveRDS(here("data", "datasets", 
                                          "mouse", "GSE140286",
                           "data_for_heatmap_12mths.rds"))
annot_for_heatmap_12mths %>% saveRDS(here("data", "datasets", 
                                          "mouse", "GSE140286",
                           "annot_for_heatmap_12mths.rds"))
```

- PCA

```{r}
PCAplot_allSamples %>% export::graph2pdf(here("data", "datasets", 
                                              "mouse", "GSE140286", 
                                              "fig", "PC1_pca.pdf"))
```

- Boxplots

```{r}
PC1_plot %>% export::graph2pdf(here("data", "datasets", 
                                              "mouse", "GSE140286", 
                                              "fig", "PC1_boxplots.pdf"),
                               height = 15, width = 11)
```

