---
title: "PathwayPCA on Antonell et al dataset"
author: "Nhi Hin"
date: "2020-05-29"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r include=FALSE}
library(pathwayPCA)
library(here)
library(dplyr)
library(magrittr)
library(tibble)
library(readr)
library(ggplot2)
library(stringr)
library(AnnotationHub)
theme_set(theme_bw())

save_pheatmap_pdf <- function(x, filename, width=8, height=11) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}
```


## Introduction

- *PathwayPCA* is enrichment analysis utilising advanced PCA techniques such 
as supervised PCA and AES (adaptive elastic-net sparse) PCA. While giving 
similar results to normal gene set testing, its advantage is that it is also 
able to work on a per-sample basis to see whether certain gene sets are 
more enriched in some samples etc. This is particularly important when 
considering diseases/conditions like Alzheimer's Disease where symptoms and 
etiologies are heterogeneous. 

- Aim here is to apply *PathwayPCA* to familial and sporadic AD human 
datasets to see if the enriched pathways of any subsets end up being 
similar to  the fAD-like zebrafish (or potentially the 5xFAD mouse model). 

## fAD microarray dataset 

### 1. Import gene sets

- According to the [PathwayPCA vignette](https://gabrielodom.github.io/pathwayPCA/articles/Supplement2-Importing_Data.html#overview) we need to set up a `pathwayCollection` object of the gene 
sets to be tested. Here I will use the Hallmark collection from MSigDB. 

- The `ens_h_mapped.rds` object was previously created from the Hallmark 
collection .gmt file from MSigDB (v.7.0). The entrezgene IDs were converted 
to `[HuGene-1_1-st] Affymetrix Human Gene 1.1 ST 
Array [transcript (gene) version]` probeset IDs as the DE analysis for 
this particular dataset was done at the probeset level (many probes 
corresponded to multiple genes). 

```{r}
hallmarkGenes <- readRDS(here("data", "datasets", "fad", "genesets",
                              "ens_h_mapped.rds"))
keggGenes <- readRDS(here("data", "datasets", "fad", 
                          "genesets", "kegg_mapped.rds"))
ireGenes <- readRDS(here("data", "ireGenes", "human", "ireProbes.rds"))
ireNames <- names(ireGenes)

hallmarkGenes <- append(hallmarkGenes, ireGenes)
keggGenes <- append(keggGenes, ireGenes)

keggNames <- names(keggGenes)
hallmarkNames <- names(hallmarkGenes)



hallmarkCol <- CreatePathwayCollection(
  sets_ls = hallmarkGenes,
  TERMS = hallmarkNames
)

keggCol <- CreatePathwayCollection(
  sets_ls = keggGenes,
  TERMS = keggNames
)

ireCol <- CreatePathwayCollection(
  sets_ls = ireGenes, TERMS = ireNames
)
```

### 2. Tidy sample/expression matrix

- The expression data must be in tidy format for inputting into 
*PathwayPCA*. 

```{r}
exprs <- readRDS(here("data", "datasets", "fad", "procData.rds"))$exprs
rownames(exprs) <- gsub(x = rownames(exprs), pattern = "_.*$",
                        replacement = "")
#samples <- readRDS(here("data", "datasets", "fad", "procData.rds"))$samples
samples <- read_csv(here("data", "datasets", "fad", "samples.csv"))
samples %<>% set_rownames(.$geo_accession) %>% rownames_to_column("Sample")
samples %<>% mutate(hasAD = c(rep(FALSE, 7), rep(TRUE, 14)))

tidyDat <- exprs %>% log2 %>%
  rownames_to_column("gene") %>% 
  #mutate(gene = gsub(x = gene, pattern = "(^.*$)", replacement = "Gene\\1")) %>%
  TransposeAssay()


# tidyDat$Sample == samples$Sample  # Check ordering is OK
# 
# tidyDat <- inner_join(samples, tidyDat, by ="Sample")

tidyDat[1:5, 1:20]  # preview
```

```{r}
initial_pca <- exprs %>% log2 %>% t %>% prcomp()

summary(initial_pca) # PC1; 97.1%, PC2 1%

initial_pca$x %>%
  as.data.frame %>%
  dplyr::select(PC1, PC2) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = PC1, y = PC2, colour = group)) +
  geom_point(size = 3, alpha = 0.6) +
  scale_colour_manual(values = c("#999999", "orangered", "turquoise"))+
  labs(x = "Principal Component 1 (37.8%)", 
       y = "Principal Component 2 (9.7%)",
       colour = "Group")

alzGenes <- keggGenes$KEGG_ALZHEIMERS_DISEASE
initial_pca_alz <- exprs %>% log2 %>% as.data.frame %>%
  rownames_to_column("probe") %>%
  dplyr::filter(probe %in% alzGenes) %>%
  column_to_rownames("probe") %>%
  t %>% prcomp

initial_pca_alz$x %>%
  as.data.frame %>%
  dplyr::select(PC1, PC2) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = PC1, y = PC2, colour = group)) +
  geom_point(size = 3, alpha = 0.6) +
  scale_colour_manual(values = c("#999999", "orangered", "turquoise")) +
    ggrepel::geom_text_repel(aes(label = Sample),size=3) +
  ggtitle("Original PCA (KEGG Alzheimer's Pathway Genes)")+
  labs(x = "Principal Component 1 (65.5%)", 
       y = "Principal Component 2 (8.3%)")
```


### 3. `Omics` data object

- The samples/expression tidy data and gene sets must now be placed inside an 
`Omics` object.

- We have to create different `Omics` objects to test different contrasts. 
*PathwayPCA* does not support complex designs like regular gene set testing 
methods utilising *limma* objects. However, they do support the following 
modes:

    - Regression on a continuous variable (e.g. age)
    - Categorical (only binary supported) (e.g. `hasPSEN` mutation)
    - Survival (not relevant to our analysis). 

- Because of this *PathwayPCA* should be able to give us some complementary 
information/results. 

### 4. AES-PCA

**Age of death**

- No pathways are significantly associated with age of death however this 
doesn't take into account whether they have AD or not.

```{r message=FALSE}
# To create a regression omics object, need a dataframe with 
# Sample and response as column names
# Here the age of death in years is the response
samples_reg <- samples %>% 
  dplyr::select(Sample, age) %>%
  dplyr::rename(response = age)

# Create omics object
omicsDatReg <- CreateOmics(
  assayData_df = tidyDat, 
  pathwayCollection_ls = hallmarkCol,
  response = samples_reg,
  respType = "reg"
)

# Perform enrichment testing
testReg <- AESPCA_pVals(
  object = omicsDatReg,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

# Results
testReg$pVals_df
```

**Has AD**

- Here we test whether pathways are enriched on the binary variable `hasAD` 
using AES-PCA

```{r}
samples_AD <- samples %>%
  dplyr::select(Sample, hasAD) %>%
  dplyr::rename(response = hasAD)

omics_AD <- CreateOmics(
  assayData_df = tidyDat,
  pathwayCollection_ls = hallmarkCol,
  response = samples_AD,
  respType = "categ"
)

# Perform enrichment testing using AES-PCA
test_AD <- AESPCA_pVals(
  object = omics_AD,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

test_AD <- readRDS(here("data", "datasets", "fad", "test_AD_fAD.rds"))

test_AD$pVals_df 

sigSets <- test_AD$pVals_df %>%
  dplyr::filter(FWER_Holm < 0.05)

sigSets

# Perform enrichment testing using Supervised PCA
# test_AD2 <- SuperPCA_pVals(
#   object = omics_AD,
#   numPCs = 2,
#   parallel = TRUE,
#   numCores = 2,
#   adjustpValues = TRUE,
#   adjustment = c("SidakSS", "BY")
# )
# test_AD2$pVals_df
```

- With KEGG instead 

```{r}
omics_AD_k <- CreateOmics(
  assayData_df = tidyDat,
  pathwayCollection_ls = keggCol,
  response = samples_AD,
  respType = "categ"
)

# Perform enrichment testing using AES-PCA
test_AD_k <- AESPCA_pVals(
  object = omics_AD_k,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

test_AD_k$pVals_df 

sigSets_k <- test_AD$pVals_df %>%
  dplyr::filter(FWER_Holm < 0.05)

sigSets_k
```


```{r}
omics_AD_ire <- CreateOmics(
  assayData_df = tidyDat,
  pathwayCollection_ls = ireCol,
  response = samples_AD,
  respType = "categ"
)
omics_AD_ire %>% saveRDS(here("data", "datasets", "fad", "omics_AD_ire.rds"))

# Perform enrichment testing using AES-PCA
test_AD_ire <- AESPCA_pVals(
  object = omics_AD_ire,
  numReps = 0,
  numPCs = 2,
  parallel = TRUE,
  numCores = 2,
  adjustpValues = TRUE,
  adjustment = c("Holm", "BH")
)

test_AD_ire <- readRDS(here("data", "datasets", "fad", "test_AD_ire.rds"))
test_AD_ire$pVals_df 

sigSets_ire <- test_AD_ire$pVals_df %>%
  dplyr::filter(FWER_Holm < 0.05)

sigSets_ire
```


### 5. Plots

- Violin plot

```{r}
ire_comparisons <- list(c("TRUE", "FALSE"))
violin_ire_plot <- test_AD_ire$PCs_ls %>%
  lapply(function(x){
    x %>% set_rownames(attributes(test_AD_ire$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "geneset") %>%
  left_join(samples, by = "Sample") %>%
  ggpubr::ggviolin(x = "hasAD", y = "V1",
                    fill = "hasAD", add = "boxplot",
                   add.params = list(fill = "white"),
                   palette = c("#dddddd", "orangered")) +
  ggpubr::stat_compare_means(comparisons = ire_comparisons)+
  #ggpubr::stat_compare_means()  +
  facet_wrap(~geneset, scales = "free_y") +
  labs(y = "PC1 Value")

violin_ire_plot %>% export::graph2pdf(here("data", "datasets", 
                                           "fad", "fig", 
                                           "violin_ire_plot"))
```


 - **Reactive oxygen species pathway**: Samples with AD tend to have lower 
 expression of genes in this pathway compared to samples without AD. 

```{r}



test_AD_ire$PCs_ls$ire3_all %>% 
  set_rownames(attributes(test_AD_ire$PCs_ls)$sampleIDs) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = hasAD)) + 
  geom_histogram(bins = 8, colour = "white") +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("#bbbbbb", "orangered")) +
  ggtitle("Human - All 3' IRE")

test_AD_ire$PCs_ls$ire3_hq %>% 
  set_rownames(attributes(test_AD_ire$PCs_ls)$sampleIDs) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = hasAD)) + 
  geom_histogram(bins = 8, colour = "white") +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("#bbbbbb", "orangered")) +
  ggtitle("Human - High-quality 3' IRE")

test_AD_ire$PCs_ls$ire5_all %>% 
  set_rownames(attributes(test_AD_ire$PCs_ls)$sampleIDs) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = hasAD)) + 
  geom_histogram(bins = 8, colour = "white", position = "identity", alpha=0.5) +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("#bbbbbb", "orangered")) +
  ggtitle("Human - All 5' IRE")

test_AD_ire$PCs_ls$ire5_hq %>% 
  set_rownames(attributes(test_AD_ire$PCs_ls)$sampleIDs) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = hasAD)) + 
  geom_histogram(bins = 8, colour = "white", position = "identity", alpha=0.5) +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("#bbbbbb", "orangered")) +
  ggtitle("Human - High quality 5' IRE")
```

- All significant gene sets

```{r eval=FALSE}
test_AD$PCs_ls[sigSets$pathways] %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_AD$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::mutate(pathway = gsub(x = pathway,
                               pattern = "_",
                               replacement = " "),
                pathway = gsub(x = pathway,
                               pattern = "HALLMARK ",
                               replacement = "")) %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = V1, fill = hasAD)) + 
  geom_histogram(bins = 8, colour = "white") +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("#bbbbbb", "orangered")) +
  facet_wrap(~pathway, scales = "free_y")
  
```

```{r}
plot_allGenesets <- test_AD$PCs_ls[sigSets$pathways] %>% 
  lapply(function(x){
    x %>% set_rownames(attributes(test_AD$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::mutate(pathway = gsub(x = pathway,
                               pattern = "_",
                               replacement = " "),
                pathway = gsub(x = pathway,
                               pattern = "HALLMARK ",
                               replacement = "")) %>%
  left_join(samples, by = "Sample") %>%
  #ggplot(aes(x = V1, fill = hasAD)) + 
  ggplot(aes(x = hasAD, y = V1, fill = hasAD)) +
  geom_boxplot() +
  #geom_histogram(bins = 8, position = "identity", alpha =0.5) +
  labs(x = "PC1 value for each sample",
       y = "Count", 
       fill = "Has AD?") +
  scale_fill_manual(values = c("cornflowerblue", "orangered")) +
  facet_wrap(~pathway, scales = "free_y")
plot_allGenesets

export::graph2pdf(plot_allGenesets, here("data", "datasets",
                                         "fad", "fig", "plot_allGenesets"),
                  width = 15, height=15)
```

## 6. Patient clustering on PC1

- Get the PC1 values into wide format 

```{r}
PC1_allSamples <- test_AD$PCs_ls %>%
  lapply(function(x){
    x %>% set_rownames(attributes(test_AD$PCs_ls)$sampleIDs) %>%
      rownames_to_column("Sample")
  }) %>%
  bind_rows(.id = "pathway") %>%
  dplyr::select(-V2) %>%
  dcast(Sample ~ pathway, value.var = "V1") %>% 
  column_to_rownames("Sample")

head(PC1_allSamples)


```

- Perform PCA to see if there's any structure / grouping in the data

```{r}
PCA_allSamples <- PC1_allSamples %>% prcomp()

PCAplot_allSamples <- PCA_allSamples$x %>%
  magrittr::extract(, c("PC1", "PC2")) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample",
                        "sex",
                        "age",
                        "apoe",
                        "psen", 
                        "group", 
                        "hasPsenMutation",
                        "hasAD")], by ="Sample") %>%
  ggplot(aes(x = PC1, y = PC2,
             colour = group,
             shape = sex)) + 
  geom_point(alpha = 0.6, size = 3) +
  scale_shape_manual(values = c(15,16, 17))+
  ggrepel::geom_text_repel(aes(label = Sample), size=3) +
  scale_colour_manual(values = c("#999999", "orangered", "turquoise")) +
  theme(aspect.ratio = 1) +
    labs(x = "Principal Component 1 (85.3%)",
         y = "Principal Component 2 (9.8%)", 
         colour = "Group", shape = "Sex") +
  ggtitle("PC1 PCA")

PCAplot_allSamples

# PCAplot_allSamples %>% export::graph2pdf(here("data","datasets",
#                                              "fad", "figs",
#                                              "pc1_pca.pdf"))
```

## 7. Heatmap


```{r}
# format data
data_for_heatmap <- PC1_allSamples %>%
  rownames_to_column("Sample") %>%
  left_join(samples[, c("Sample", "hasAD", "apoe", "group", "age")])%>%
  dplyr::arrange(hasAD, apoe) 

annot_for_heatmap <- data_for_heatmap %>%
  as.data.frame %>%
  dplyr::select(Sample, hasAD, apoe, group, age) %>%
  dplyr::mutate(hasAD = case_when(
    hasAD == TRUE ~ "AD",
    hasAD == FALSE ~ "Control"
  ),
  age = as.numeric(age))%>%
  dplyr::arrange(hasAD, group)%>%
  column_to_rownames("Sample")
  

data_for_heatmap %<>% 
    column_to_rownames("Sample") %>%
  dplyr::select(-group, -age, -apoe, -hasAD) %>%
  t%>%
  set_rownames(gsub(x = rownames(.),
                              pattern = "HALLMARK_",
                              replacement = "") )

palette <- colorRampPalette(c("#10c1e5", "#82e0b4",
                              "#F9F9F9", "#FBB829", "#FF0066"))(100)

plot_heatmap <- pheatmap::pheatmap(data_for_heatmap,
                   cluster_cols = FALSE,
                   scale="row",
                   color = palette,
                   annotation_col = annot_for_heatmap,
                   border_color = "white"
                   )

plot_heatmap %>% save_pheatmap_pdf(filename = here("data","datasets",
                                        "fad", "fig","heatmap2.pdf"))
```

## 8. Check if the outlier samples are actually mislabeled

- Probe-gene annotations used in the analysis of this Antonell dataset. 

```{r}
annot <- readRDS(here("data", "datasets", "fad", "annot.rds")) 
```

- Get chromosome data for these 

```{r}
ah <- AnnotationHub()
# ah %>%
#   subset(grepl("sapiens", species)) %>%
#   subset(rdataclass == "EnsDb")
ensDb <- ah[["AH69187"]]
genes <- genes(ensDb)
genes <- as.data.frame(genes)
head(genes)

chrInfo <- genes %>% dplyr::select(gene_id, seqnames, gene_biotype) %>%
  dplyr::filter(gene_biotype == "protein_coding") %>%
  dplyr::distinct(gene_id, seqnames, .keep_all = TRUE) %>%
  dplyr::select(-gene_biotype)
```

- Extract the sex-specific genes located on the X and Y chromosomes. 

```{r}
annot %<>% left_join(chrInfo, by = c("ENSEMBL"="gene_id")) 

x_genes <- annot %>% dplyr::filter(seqnames == "X")
y_genes <- annot %>% dplyr::filter(seqnames == "Y")

```

- PCA of Y genes

```{r}
ygene_pca <- exprs %>% as.data.frame %>%
  rownames_to_column("gene") %>%
  dplyr::filter(gene %in% y_genes$PROBEID) %>%
  column_to_rownames("gene") %>%
  log2  %>% t %>% prcomp()

rownames(ygene_pca$x) <- gsub(x = rownames(ygene_pca$x),
                              pattern = "_.*$", replacement = "")


summary(ygene_pca) # PC1; 83.5%, PC2 8.6%

ygene_pca$x %>%
  as.data.frame %>%
  dplyr::select(PC1, PC2) %>%
  rownames_to_column("Sample") %>%
  left_join(samples, by = "Sample") %>%
  ggplot(aes(x = PC1, y = PC2, colour = group, shape = sex)) +
  ggrepel::geom_text_repel(aes(label = Sample), hjust=0, vjust=0) +
  geom_point(size = 3, alpha = 0.6) +
  scale_colour_manual(values = c("#999999", "orangered", "turquoise"))
```

- Clear separation on PCA between male and female samples. 

- From this PCA, the two outlier samples (in terms of overall PC1 expression) 
`GSM967918` (labelled female control), and `GSM967927` (labelled male AD) do 
seem to have the right sex labelled. 


## z-scores

- Calculate the z-scores for the PC1 values

```{r}
PC1_allSamples %>%
  sapply(function(x){
    x %>% summary
  })

scaled_PC1 <- PC1_allSamples %>%
  sapply(function(x){
    x <- (x-mean(x))/sd(x) # Calculate x-score for each Hallmark geneset
  }) %>%
  set_rownames(rownames(PC1_allSamples))

scaled_PC1 %<>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(annot_for_heatmap[,c("hasAD","group","apoe")] %>%
              rownames_to_column("Sample")) %>%
  arrange(hasAD,group,apoe) %>%
  dplyr::select(-hasAD,-group,-apoe) %>%
  column_to_rownames("Sample")

```

- Import in sAD data (Temporal Cortex)

```{r}
PC1_sAD <- readRDS(here("data", "datasets", "mayo", 
                               "PC1_TCsamples.rds")) %>%
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  left_join(heatmapAnnot_sAD[,c("Diagnosis", "Cluster")] %>% 
              rownames_to_column("Sample"))%>%
  arrange(Cluster, Diagnosis) %>%
  dplyr::select(-Diagnosis, -Cluster)%>%
  column_to_rownames("Sample")

heatmapAnnot_sAD <- readRDS(here("data", "datasets", "mayo", 
                                 "heatmapAnnot.rds"))%>%
  dplyr::select(Diagnosis, AgeAtDeath,ApoE, Cluster)

scaled_PC1_sAD <- PC1_sAD %>%
  sapply(function(x){
    x <- (x-mean(x))/sd(x) # Calculate x-score for each Hallmark geneset
  }) %>%
  set_rownames(rownames(PC1_sAD)) %>% #Rearrange order
  magrittr::extract(rownames(heatmapAnnot_sAD),)
```

- Attempt 1: Just joining the data and YOLO heatmapping it

```{r}
combinedPC1 <- rbind(scaled_PC1, scaled_PC1_sAD)
annot_for_heatmap2 <-annot_for_heatmap %>%rownames_to_column("temp") %>%
  dplyr::mutate(Cluster = 6,
                apoe = gsub(x = apoe, pattern = "_", replacement="")) %>%
  dplyr::rename(Diagnosis = group,
                ApoE = apoe,
                AgeAtDeath = age)%>%
  dplyr::select(temp,Diagnosis,AgeAtDeath,ApoE,Cluster) %>%
  column_to_rownames("temp")

combinedAnnot <- rbind(annot_for_heatmap2, heatmapAnnot_sAD)

combinedHeatmap <- pheatmap(combinedPC1%>%t,
         color = palette,
         cluster_cols = FALSE,
         annotation_col = combinedAnnot)

# save_pheatmap_pdf(combinedHeatmap, here("data", "datasets",
#                                         "fad", "fig",
#                                         "combinedHeatmap.pdf"),
#                     width = 15, height=11)
```


```{r}
scaled_PC1

pheatmap(scaled_PC1,
         color=palette)

pheatmap::pheatmap(scaled_PC1,
                   cluster_cols = FALSE,
                   scale="row",
                   color = palette,
                   annotation_col = annot_for_heatmap,
                   border_color = "white"
                   )
```



## Export Objects

- Sample data

```{r}
samples %>% saveRDS(here("data", "datasets", "fad", "samples.rds"))
```


- PC1 values

```{r eval=FALSE}
PC1_allSamples %>% saveRDS(here("data", "datasets", 
                                "fad", "PC1_allSamples.rds"))

scaled_PC1 %>% saveRDS(here("data", "datasets", "fad", "scaled_PC1.rds"))
```

- PC1 PCA

```{r eval=FALSE}
PCAplot_allSamples %>% saveRDS(here("data", "datasets", "fad", 
                                    "PC1_pca.rds"))

PCAplot_allSamples %>% export::graph2pdf(here("data", "datasets", "fad", "fig",
                                    "PC1_pca.pdf"))
```


- Heatmap

```{r eval=FALSE}
save_pheatmap_pdf(plot_heatmap, here("data", "datasets", "fad", "fig", 
                                     "heatmap.pdf"), 
                  width = 11, height = 11)
```



## todo

- Check x and y gene expression to see if we can relabel samples

- Boxplots instead of histograms

- PC1 value turn into z-score (subtract mean divide s.d.) for each dataset and pathway. 
Compare this standardised score between datasets. 

- Check predictive variables within clusters (ApoE stacked boxplot, age of death as boxplot)













